// @aws-sdk/client-qbusiness/dist-cjs/index.js
'use strict';

var middlewareEventstream = require('@aws-sdk/middleware-eventstream');
var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var eventstreamSerdeConfigResolver = require('@smithy/eventstream-serde-config-resolver');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "qbusiness",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class QBusinessClient extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = eventstreamSerdeConfigResolver.resolveEventStreamSerdeConfig(_config_6);
        const _config_8 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_7);
        const _config_9 = middlewareEventstream.resolveEventStreamConfig(_config_8);
        const _config_10 = resolveRuntimeExtensions(_config_9, configuration?.extensions || []);
        this.config = _config_10;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultQBusinessHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

let QBusinessServiceException$1 = class QBusinessServiceException extends smithyClient.ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, QBusinessServiceException.prototype);
    }
};

let AccessDeniedException$1 = class AccessDeniedException extends QBusinessServiceException$1 {
    name = "AccessDeniedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
    }
};
let ConflictException$1 = class ConflictException extends QBusinessServiceException$1 {
    name = "ConflictException";
    $fault = "client";
    resourceId;
    resourceType;
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
        this.resourceId = opts.resourceId;
        this.resourceType = opts.resourceType;
    }
};
let InternalServerException$1 = class InternalServerException extends QBusinessServiceException$1 {
    name = "InternalServerException";
    $fault = "server";
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalServerException.prototype);
    }
};
let ResourceNotFoundException$1 = class ResourceNotFoundException extends QBusinessServiceException$1 {
    name = "ResourceNotFoundException";
    $fault = "client";
    resourceId;
    resourceType;
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
        this.resourceId = opts.resourceId;
        this.resourceType = opts.resourceType;
    }
};
let ServiceQuotaExceededException$1 = class ServiceQuotaExceededException extends QBusinessServiceException$1 {
    name = "ServiceQuotaExceededException";
    $fault = "client";
    resourceId;
    resourceType;
    constructor(opts) {
        super({
            name: "ServiceQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceQuotaExceededException.prototype);
        this.resourceId = opts.resourceId;
        this.resourceType = opts.resourceType;
    }
};
let ThrottlingException$1 = class ThrottlingException extends QBusinessServiceException$1 {
    name = "ThrottlingException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ThrottlingException.prototype);
    }
};
let ValidationException$1 = class ValidationException extends QBusinessServiceException$1 {
    name = "ValidationException";
    $fault = "client";
    reason;
    fields;
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ValidationException.prototype);
        this.reason = opts.reason;
        this.fields = opts.fields;
    }
};
let ExternalResourceException$1 = class ExternalResourceException extends QBusinessServiceException$1 {
    name = "ExternalResourceException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ExternalResourceException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ExternalResourceException.prototype);
    }
};
let LicenseNotFoundException$1 = class LicenseNotFoundException extends QBusinessServiceException$1 {
    name = "LicenseNotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "LicenseNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LicenseNotFoundException.prototype);
    }
};
let MediaTooLargeException$1 = class MediaTooLargeException extends QBusinessServiceException$1 {
    name = "MediaTooLargeException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "MediaTooLargeException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, MediaTooLargeException.prototype);
    }
};

const _A = "Application";
const _AAC = "AppliedAttachmentsConfiguration";
const _AC = "AccessConfiguration";
const _ACL = "ActionConfigurationList";
const _ACMC = "AppliedCreatorModeConfiguration";
const _ACR = "AuthChallengeRequest";
const _ACRE = "AuthChallengeRequestEvent";
const _ACREu = "AuthChallengeResponseEvent";
const _ACRu = "AuthChallengeResponse";
const _ACc = "AccessControl";
const _ACcc = "AccessControls";
const _ACct = "ActionConfiguration";
const _ACt = "AttachmentsConfiguration";
const _ADE = "AccessDeniedException";
const _AE = "ActionExecution";
const _AEC = "AudioExtractionConfiguration";
const _AEE = "ActionExecutionEvent";
const _AEP = "ActionExecutionPayload";
const _AEPF = "ActionExecutionPayloadField";
const _AF = "AttributeFilter";
const _AFC = "ActionFilterConfiguration";
const _AFt = "AttributeFilters";
const _AG = "AssociatedGroup";
const _AGs = "AssociatedGroups";
const _AI = "AttachmentInput";
const _AIE = "AttachmentInputEvent";
const _AIt = "AttachmentsInput";
const _AL = "AttachmentList";
const _AO = "AttachmentOutput";
const _AOC = "AppliedOrchestrationConfiguration";
const _AOt = "AttachmentsOutput";
const _AP = "AssociatePermission";
const _APIS = "APISchema";
const _APR = "AssociatePermissionRequest";
const _APRs = "AssociatePermissionResponse";
const _AR = "ActionReview";
const _ARE = "ActionReviewEvent";
const _ARP = "ActionReviewPayload";
const _ARPF = "ActionReviewPayloadField";
const _ARPFAV = "ActionReviewPayloadFieldAllowedValue";
const _ARPFAVc = "ActionReviewPayloadFieldAllowedValues";
const _AS = "ActionSummary";
const _ASC = "AutoSubscriptionConfiguration";
const _ASD = "AudioSourceDetails";
const _AU = "AssociatedUser";
const _AUs = "AssociatedUsers";
const _Ac = "Actions";
const _Ap = "Applications";
const _At = "Attachment";
const _BAC = "BasicAuthConfiguration";
const _BDD = "BatchDeleteDocument";
const _BDDR = "BatchDeleteDocumentRequest";
const _BDDRa = "BatchDeleteDocumentResponse";
const _BEC = "BrowserExtensionConfiguration";
const _BPC = "BlockedPhrasesConfiguration";
const _BPCU = "BlockedPhrasesConfigurationUpdate";
const _BPD = "BatchPutDocument";
const _BPDR = "BatchPutDocumentRequest";
const _BPDRa = "BatchPutDocumentResponse";
const _C = "Conversation";
const _CA = "CreateApplication";
const _CAR = "CreateApplicationRequest";
const _CARr = "CreateApplicationResponse";
const _CAWEU = "CreateAnonymousWebExperienceUrl";
const _CAWEUR = "CreateAnonymousWebExperienceUrlRequest";
const _CAWEURr = "CreateAnonymousWebExperienceUrlResponse";
const _CBR = "ContentBlockerRule";
const _CC = "CustomizationConfiguration";
const _CCRC = "CreateChatResponseConfiguration";
const _CCRCR = "CreateChatResponseConfigurationRequest";
const _CCRCRr = "CreateChatResponseConfigurationResponse";
const _CDA = "CheckDocumentAccess";
const _CDAR = "CheckDocumentAccessRequest";
const _CDARh = "CheckDocumentAccessResponse";
const _CDARr = "CreateDataAccessorRequest";
const _CDARre = "CreateDataAccessorResponse";
const _CDAr = "CreateDataAccessor";
const _CDS = "CreateDataSource";
const _CDSR = "CreateDataSourceRequest";
const _CDSRr = "CreateDataSourceResponse";
const _CE = "ConfigurationEvent";
const _CEo = "ConflictException";
const _CFS = "CopyFromSource";
const _CI = "ChatInput";
const _CIR = "CreateIndexRequest";
const _CIRr = "CreateIndexResponse";
const _CIS = "ChatInputStream";
const _CIr = "CreateIndex";
const _CMC = "CreatorModeConfiguration";
const _CMCh = "ChatModeConfiguration";
const _CO = "ChatOutput";
const _COS = "ChatOutputStream";
const _CP = "CreatePlugin";
const _CPC = "CustomPluginConfiguration";
const _CPR = "CreatePluginRequest";
const _CPRr = "CreatePluginResponse";
const _CR = "CreateRetriever";
const _CRC = "ChatResponseConfiguration";
const _CRCD = "ChatResponseConfigurationDetail";
const _CRCh = "ChatResponseConfigurations";
const _CRR = "ContentRetrievalRule";
const _CRRr = "CreateRetrieverRequest";
const _CRRre = "CreateRetrieverResponse";
const _CS = "ConversationSource";
const _CSI = "ChatSyncInput";
const _CSO = "ChatSyncOutput";
const _CSR = "CancelSubscriptionRequest";
const _CSRa = "CancelSubscriptionResponse";
const _CSRr = "CreateSubscriptionRequest";
const _CSRre = "CreateSubscriptionResponse";
const _CSa = "CancelSubscription";
const _CSh = "ChatSync";
const _CSo = "ContentSource";
const _CSr = "CreateSubscription";
const _CU = "CreateUser";
const _CUR = "CreateUserRequest";
const _CURr = "CreateUserResponse";
const _CWE = "CreateWebExperience";
const _CWER = "CreateWebExperienceRequest";
const _CWERr = "CreateWebExperienceResponse";
const _Ch = "Chat";
const _Co = "Conversations";
const _D = "Document";
const _DA = "DataAccessor";
const _DAAC = "DataAccessorAuthenticationConfiguration";
const _DAAD = "DataAccessorAuthenticationDetail";
const _DABC = "DateAttributeBoostingConfiguration";
const _DABCo = "DocumentAttributeBoostingConfiguration";
const _DABOM = "DocumentAttributeBoostingOverrideMap";
const _DAC = "DocumentAclCondition";
const _DACo = "DocumentAttributeCondition";
const _DACoc = "DocumentAttributeConfiguration";
const _DACocu = "DocumentAclConditions";
const _DACocum = "DocumentAttributeConfigurations";
const _DAG = "DocumentAclGroup";
const _DAGo = "DocumentAclGroups";
const _DAITTIC = "DataAccessorIdcTrustedTokenIssuerConfiguration";
const _DAM = "DocumentAclMembership";
const _DAN = "DataAccessorName";
const _DAR = "DeleteApplicationRequest";
const _DARe = "DeleteApplicationResponse";
const _DARel = "DeleteAttachmentRequest";
const _DARele = "DeleteAttachmentResponse";
const _DAT = "DocumentAttributeTarget";
const _DAU = "DocumentAclUser";
const _DAUo = "DocumentAclUsers";
const _DAV = "DocumentAttributeValue";
const _DAa = "DataAccessors";
const _DAe = "DeleteApplication";
const _DAel = "DeleteAttachment";
const _DAo = "DocumentAcl";
const _DAoc = "DocumentAttribute";
const _DAocu = "DocumentAttributes";
const _DC = "DocumentContent";
const _DCCC = "DeleteChatControlsConfiguration";
const _DCCCR = "DeleteChatControlsConfigurationRequest";
const _DCCCRe = "DeleteChatControlsConfigurationResponse";
const _DCR = "DeleteConversationRequest";
const _DCRC = "DeleteChatResponseConfiguration";
const _DCRCR = "DeleteChatResponseConfigurationRequest";
const _DCRCRe = "DeleteChatResponseConfigurationResponse";
const _DCRe = "DeleteConversationResponse";
const _DCe = "DeleteConversation";
const _DD = "DeleteDocument";
const _DDA = "DeleteDataAccessor";
const _DDAR = "DeleteDataAccessorRequest";
const _DDARe = "DeleteDataAccessorResponse";
const _DDL = "DocumentDetailList";
const _DDS = "DeleteDataSource";
const _DDSR = "DeleteDataSourceRequest";
const _DDSRe = "DeleteDataSourceResponse";
const _DDe = "DeleteDocuments";
const _DDo = "DocumentDetails";
const _DEC = "DocumentEnrichmentConfiguration";
const _DG = "DeleteGroup";
const _DGR = "DeleteGroupRequest";
const _DGRe = "DeleteGroupResponse";
const _DI = "DeleteIndex";
const _DIR = "DeleteIndexRequest";
const _DIRe = "DeleteIndexResponse";
const _DP = "DeletePlugin";
const _DPR = "DeletePluginRequest";
const _DPRe = "DeletePluginResponse";
const _DPRi = "DisassociatePermissionRequest";
const _DPRis = "DisassociatePermissionResponse";
const _DPi = "DisassociatePermission";
const _DR = "DeleteRetriever";
const _DRR = "DeleteRetrieverRequest";
const _DRRe = "DeleteRetrieverResponse";
const _DS = "DataSource";
const _DSSJ = "DataSourceSyncJob";
const _DSSJM = "DataSourceSyncJobMetrics";
const _DSSJa = "DataSourceSyncJobs";
const _DSVC = "DataSourceVpcConfiguration";
const _DSa = "DataSources";
const _DU = "DeleteUser";
const _DUR = "DeleteUserRequest";
const _DURe = "DeleteUserResponse";
const _DWE = "DeleteWebExperience";
const _DWER = "DeleteWebExperienceRequest";
const _DWERe = "DeleteWebExperienceResponse";
const _Do = "Documents";
const _EC = "EncryptionConfiguration";
const _ED = "ErrorDetail";
const _EDS = "EligibleDataSource";
const _EDSl = "EligibleDataSources";
const _EOIE = "EndOfInputEvent";
const _ERE = "ExternalResourceException";
const _FAE = "FailedAttachmentEvent";
const _FD = "FailedDocument";
const _FDa = "FailedDocuments";
const _GA = "GetApplication";
const _GAR = "GetApplicationRequest";
const _GARe = "GetApplicationResponse";
const _GCCC = "GetChatControlsConfiguration";
const _GCCCR = "GetChatControlsConfigurationRequest";
const _GCCCRe = "GetChatControlsConfigurationResponse";
const _GCRC = "GetChatResponseConfiguration";
const _GCRCR = "GetChatResponseConfigurationRequest";
const _GCRCRe = "GetChatResponseConfigurationResponse";
const _GDA = "GetDataAccessor";
const _GDAR = "GetDataAccessorRequest";
const _GDARe = "GetDataAccessorResponse";
const _GDC = "GetDocumentContent";
const _GDCR = "GetDocumentContentRequest";
const _GDCRe = "GetDocumentContentResponse";
const _GDS = "GetDataSource";
const _GDSR = "GetDataSourceRequest";
const _GDSRe = "GetDataSourceResponse";
const _GG = "GetGroup";
const _GGR = "GetGroupRequest";
const _GGRe = "GetGroupResponse";
const _GI = "GetIndex";
const _GIR = "GetIndexRequest";
const _GIRe = "GetIndexResponse";
const _GM = "GroupMembers";
const _GMR = "GetMediaRequest";
const _GMRe = "GetMediaResponse";
const _GMe = "GetMedia";
const _GP = "GetPlugin";
const _GPR = "GetPluginRequest";
const _GPRe = "GetPluginResponse";
const _GPRet = "GetPolicyRequest";
const _GPReto = "GetPolicyResponse";
const _GPe = "GetPolicy";
const _GR = "GetRetriever";
const _GRR = "GetRetrieverRequest";
const _GRRe = "GetRetrieverResponse";
const _GS = "GroupSummary";
const _GSD = "GroupStatusDetail";
const _GSDr = "GroupStatusDetails";
const _GSL = "GroupSummaryList";
const _GU = "GetUser";
const _GUR = "GetUserRequest";
const _GURe = "GetUserResponse";
const _GWE = "GetWebExperience";
const _GWER = "GetWebExperienceRequest";
const _GWERe = "GetWebExperienceResponse";
const _HC = "HookConfiguration";
const _HRC = "HallucinationReductionConfiguration";
const _I = "Index";
const _IAC = "IdcAuthConfiguration";
const _IC = "InstructionCollection";
const _ICC = "IndexCapacityConfiguration";
const _IDEC = "InlineDocumentEnrichmentConfiguration";
const _IDECn = "InlineDocumentEnrichmentConfigurations";
const _IEC = "ImageExtractionConfiguration";
const _IPC = "IdentityProviderConfiguration";
const _IS = "IndexStatistics";
const _ISD = "ImageSourceDetails";
const _ISE = "InternalServerException";
const _In = "Indices";
const _KIC = "KendraIndexConfiguration";
const _KKI = "KmsKeyId";
const _LA = "ListApplications";
const _LAR = "ListApplicationsRequest";
const _LARi = "ListApplicationsResponse";
const _LARis = "ListAttachmentsRequest";
const _LARist = "ListAttachmentsResponse";
const _LAi = "ListAttachments";
const _LC = "ListConversations";
const _LCR = "ListConversationsRequest";
const _LCRC = "ListChatResponseConfigurations";
const _LCRCR = "ListChatResponseConfigurationsRequest";
const _LCRCRi = "ListChatResponseConfigurationsResponse";
const _LCRi = "ListConversationsResponse";
const _LD = "ListDocuments";
const _LDA = "ListDataAccessors";
const _LDAR = "ListDataAccessorsRequest";
const _LDARi = "ListDataAccessorsResponse";
const _LDR = "ListDocumentsRequest";
const _LDRi = "ListDocumentsResponse";
const _LDS = "ListDataSources";
const _LDSR = "ListDataSourcesRequest";
const _LDSRi = "ListDataSourcesResponse";
const _LDSSJ = "ListDataSourceSyncJobs";
const _LDSSJR = "ListDataSourceSyncJobsRequest";
const _LDSSJRi = "ListDataSourceSyncJobsResponse";
const _LG = "ListGroups";
const _LGR = "ListGroupsRequest";
const _LGRi = "ListGroupsResponse";
const _LI = "ListIndices";
const _LIR = "ListIndicesRequest";
const _LIRi = "ListIndicesResponse";
const _LM = "ListMessages";
const _LMR = "ListMessagesRequest";
const _LMRi = "ListMessagesResponse";
const _LNFE = "LicenseNotFoundException";
const _LP = "ListPlugins";
const _LPA = "ListPluginActions";
const _LPAR = "ListPluginActionsRequest";
const _LPARi = "ListPluginActionsResponse";
const _LPR = "ListPluginsRequest";
const _LPRi = "ListPluginsResponse";
const _LPTA = "ListPluginTypeActions";
const _LPTAR = "ListPluginTypeActionsRequest";
const _LPTARi = "ListPluginTypeActionsResponse";
const _LPTM = "ListPluginTypeMetadata";
const _LPTMR = "ListPluginTypeMetadataRequest";
const _LPTMRi = "ListPluginTypeMetadataResponse";
const _LPTMS = "ListPluginTypeMetadataSummaries";
const _LR = "ListRetrievers";
const _LRR = "ListRetrieversRequest";
const _LRRi = "ListRetrieversResponse";
const _LS = "ListSubscriptions";
const _LSR = "ListSubscriptionsRequest";
const _LSRi = "ListSubscriptionsResponse";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _LWE = "ListWebExperiences";
const _LWER = "ListWebExperiencesRequest";
const _LWERi = "ListWebExperiencesResponse";
const _M = "Message";
const _ME = "MetadataEvent";
const _MEC = "MediaExtractionConfiguration";
const _MG = "MemberGroup";
const _MGe = "MemberGroups";
const _MTLE = "MediaTooLargeException";
const _MU = "MemberUser";
const _MUF = "MessageUsefulnessFeedback";
const _MUe = "MemberUsers";
const _Me = "Messages";
const _NABC = "NumberAttributeBoostingConfiguration";
const _NAC = "NoAuthConfiguration";
const _NIC = "NativeIndexConfiguration";
const _OACCC = "OAuth2ClientCredentialConfiguration";
const _OC = "OrchestrationConfiguration";
const _OIDCPC = "OpenIDConnectProviderConfiguration";
const _P = "Payload";
const _PAC = "PluginAuthConfiguration";
const _PC = "PermissionCondition";
const _PCe = "PersonalizationConfiguration";
const _PCer = "PermissionConditions";
const _PCl = "PluginConfiguration";
const _PF = "PutFeedback";
const _PFR = "PutFeedbackRequest";
const _PG = "PrincipalGroup";
const _PGR = "PutGroupRequest";
const _PGRu = "PutGroupResponse";
const _PGu = "PutGroup";
const _PTMS = "PluginTypeMetadataSummary";
const _PU = "PrincipalUser";
const _Pl = "Plugin";
const _Plu = "Plugins";
const _Pr = "Principals";
const _Pri = "Principal";
const _QAC = "QAppsConfiguration";
const _QSC = "QuickSightConfiguration";
const _R = "Retriever";
const _RC = "RelevantContent";
const _RCL = "RelevantContentList";
const _RCS = "RetrieverContentSource";
const _RCe = "ResponseConfiguration";
const _RCes = "ResponseConfigurations";
const _RCet = "RetrieverConfiguration";
const _RCu = "RuleConfiguration";
const _RNFE = "ResourceNotFoundException";
const _Re = "Retrievers";
const _Ru = "Rule";
const _Rul = "Rules";
const _S = "S3";
const _SA = "ScoreAttributes";
const _SABC = "StringAttributeBoostingConfiguration";
const _SAo = "SourceAttribution";
const _SAou = "SourceAttributions";
const _SC = "SamlConfiguration";
const _SD = "SubscriptionDetails";
const _SDSSJ = "StartDataSourceSyncJob";
const _SDSSJR = "StartDataSourceSyncJobRequest";
const _SDSSJRt = "StartDataSourceSyncJobResponse";
const _SDSSJRto = "StopDataSourceSyncJobRequest";
const _SDSSJRtop = "StopDataSourceSyncJobResponse";
const _SDSSJt = "StopDataSourceSyncJob";
const _SDo = "SourceDetails";
const _SE = "SnippetExcerpt";
const _SLABC = "StringListAttributeBoostingConfiguration";
const _SP = "SubscriptionPrincipal";
const _SPC = "SamlProviderConfiguration";
const _SQEE = "ServiceQuotaExceededException";
const _SRC = "SearchRelevantContent";
const _SRCR = "SearchRelevantContentRequest";
const _SRCRe = "SearchRelevantContentResponse";
const _Su = "Subscription";
const _Sub = "Subscriptions";
const _T = "Tag";
const _TC = "TopicConfiguration";
const _TCo = "TopicConfigurations";
const _TDS = "TextDocumentStatistics";
const _TE = "ThrottlingException";
const _TIE = "TextInputEvent";
const _TOE = "TextOutputEvent";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _TS = "TextSegment";
const _TSL = "TextSegmentList";
const _Ta = "Tags";
const _UA = "UserAlias";
const _UAG = "UsersAndGroups";
const _UAR = "UpdateApplicationRequest";
const _UARp = "UpdateApplicationResponse";
const _UAp = "UpdateApplication";
const _UAs = "UserAliases";
const _UCCC = "UpdateChatControlsConfiguration";
const _UCCCR = "UpdateChatControlsConfigurationRequest";
const _UCCCRp = "UpdateChatControlsConfigurationResponse";
const _UCRC = "UpdateChatResponseConfiguration";
const _UCRCR = "UpdateChatResponseConfigurationRequest";
const _UCRCRp = "UpdateChatResponseConfigurationResponse";
const _UDA = "UpdateDataAccessor";
const _UDAR = "UpdateDataAccessorRequest";
const _UDARp = "UpdateDataAccessorResponse";
const _UDS = "UpdateDataSource";
const _UDSR = "UpdateDataSourceRequest";
const _UDSRp = "UpdateDataSourceResponse";
const _UI = "UpdateIndex";
const _UIR = "UpdateIndexRequest";
const _UIRp = "UpdateIndexResponse";
const _UP = "UpdatePlugin";
const _UPR = "UpdatePluginRequest";
const _UPRp = "UpdatePluginResponse";
const _UR = "UntagResource";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _URRp = "UpdateRetrieverRequest";
const _URRpd = "UpdateRetrieverResponse";
const _URp = "UpdateRetriever";
const _US = "UpdateSubscription";
const _USR = "UpdateSubscriptionRequest";
const _USRp = "UpdateSubscriptionResponse";
const _UU = "UpdateUser";
const _UUR = "UpdateUserRequest";
const _UURp = "UpdateUserResponse";
const _UWE = "UpdateWebExperience";
const _UWER = "UpdateWebExperienceRequest";
const _UWERp = "UpdateWebExperienceResponse";
const _VE = "ValidationException";
const _VEC = "VideoExtractionConfiguration";
const _VEF = "ValidationExceptionField";
const _VEFa = "ValidationExceptionFields";
const _VSD = "VideoSourceDetails";
const _WE = "WebExperience";
const _WEAC = "WebExperienceAuthConfiguration";
const _WEe = "WebExperiences";
const _a = "action";
const _aA = "applicationArn";
const _aAF = "andAllFilters";
const _aC = "accessControls";
const _aCM = "attachmentsControlMode";
const _aCR = "authChallengeResponse";
const _aCRE = "authChallengeResponseEvent";
const _aCREu = "authChallengeRequestEvent";
const _aCRu = "authChallengeRequest";
const _aCc = "actionConfigurations";
const _aCcc = "accessConfiguration";
const _aCt = "attachmentsConfiguration";
const _aCu = "authConfiguration";
const _aCut = "authenticationConfiguration";
const _aD = "authenticationDetail";
const _aE = "actionExecution";
const _aEC = "audioExtractionConfiguration";
const _aEE = "actionExecutionEvent";
const _aES = "audioExtractionStatus";
const _aET = "audioExtractionType";
const _aEt = "attachmentEvent";
const _aF = "allowedFormat";
const _aFt = "attributeFilter";
const _aI = "actionIdentifier";
const _aIJS = "arrayItemJsonSchema";
const _aIp = "applicationId";
const _aIt = "attachmentId";
const _aR = "actionReview";
const _aRE = "actionReviewEvent";
const _aS = "autoSubscribe";
const _aSC = "autoSubscriptionConfiguration";
const _aSD = "audioSourceDetails";
const _aST = "apiSchemaType";
const _aSp = "apiSchema";
const _aT = "authenticationType";
const _aU = "authorizationUrl";
const _aUn = "anonymousUrl";
const _aUu = "authenticationUrl";
const _aV = "allowedValues";
const _aVB = "attributeValueBoosting";
const _aVO = "attributeValueOperator";
const _ac = "actions";
const _acc = "access";
const _al = "allowlist";
const _ap = "applications";
const _at = "attachment";
const _att = "attachments";
const _attr = "attributes";
const _b = "body";
const _bAC = "basicAuthConfiguration";
const _bDIS = "boostingDurationInSeconds";
const _bEC = "browserExtensionConfiguration";
const _bL = "boostingLevel";
const _bO = "boostingOverride";
const _bOe = "beginOffset";
const _bP = "blockedPhrases";
const _bPCU = "blockedPhrasesConfigurationUpdate";
const _bPTCOU = "blockedPhrasesToCreateOrUpdate";
const _bPTD = "blockedPhrasesToDelete";
const _bS = "buildStatus";
const _bT = "boostingType";
const _bl = "blob";
const _bu = "bucket";
const _c = "client";
const _cA = "createdAt";
const _cAo = "containsAll";
const _cAon = "containsAny";
const _cBR = "contentBlockerRule";
const _cC = "capacityConfiguration";
const _cCSSU = "customCSSUrl";
const _cCu = "customizationConfiguration";
const _cE = "configurationEvent";
const _cF = "copyFrom";
const _cI = "conversationId";
const _cIFOIDC = "clientIdsForOIDC";
const _cIu = "customInstructions";
const _cK = "conditionKey";
const _cM = "chatMode";
const _cMC = "creatorModeControl";
const _cMCh = "chatModeConfiguration";
const _cMCr = "creatorModeConfiguration";
const _cN = "clientNamespace";
const _cNi = "citationNumber";
const _cO = "conditionOperator";
const _cPC = "customPluginConfiguration";
const _cRC = "chatResponseConfigurations";
const _cRCA = "chatResponseConfigurationArn";
const _cRCI = "chatResponseConfigurationId";
const _cRR = "contentRetrievalRule";
const _cS = "currentSubscription";
const _cSo = "contentSource";
const _cT = "clientToken";
const _cTo = "contentType";
const _cV = "conditionValues";
const _ca = "category";
const _co = "control";
const _com = "comment";
const _con = "conditions";
const _cond = "condition";
const _conf = "configuration";
const _cont = "content";
const _conv = "conversations";
const _conve = "conversation";
const _d = "description";
const _dA = "documentAcl";
const _dAA = "dataAccessorArn";
const _dAC = "documentAttributeConfigurations";
const _dAF = "documentAttributeFilter";
const _dAI = "dataAccessorId";
const _dAa = "dataAccessors";
const _dAo = "documentsAdded";
const _dAoc = "documentAttributes";
const _dC = "dateConfiguration";
const _dCO = "documentContentOperator";
const _dD = "displayDescription";
const _dDL = "documentDetailList";
const _dDo = "documentsDeleted";
const _dE = "defaultEndpoint";
const _dEC = "documentEnrichmentConfiguration";
const _dF = "documentsFailed";
const _dI = "documentId";
const _dIa = "datasourceId";
const _dL = "denyList";
const _dM = "documentsModified";
const _dN = "displayName";
const _dO = "displayOrder";
const _dS = "documentsScanned";
const _dSA = "dataSourceArn";
const _dSEC = "dataSourceErrorCode";
const _dSI = "dataSourceId";
const _dSIa = "dataSourceIds";
const _dSSI = "dataSourceSyncId";
const _dST = "defaultSubscriptionType";
const _dSa = "dataSources";
const _dT = "documentTitle";
const _dU = "documentUri";
const _dV = "displayValue";
const _dVa = "dateValue";
const _da = "data";
const _do = "documents";
const _e = "error";
const _eBE = "enabledBrowserExtensions";
const _eC = "encryptionConfiguration";
const _eCM = "exampleChatMessages";
const _eCr = "errorCode";
const _eD = "errorDetail";
const _eDS = "eligibleDataSources";
const _eI = "externalIds";
const _eIx = "executionId";
const _eM = "errorMessage";
const _eO = "endOffset";
const _eOIE = "endOfInputEvent";
const _eT = "equalsTo";
const _eTM = "endTimeMilliseconds";
const _eTn = "endTime";
const _eUAG = "excludedUsersAndGroups";
const _ex = "examples";
const _f = "fields";
const _fA = "failedAttachments";
const _fAE = "failedAttachmentEvent";
const _fC = "filterConfiguration";
const _fD = "failedDocuments";
const _fS = "fileSize";
const _fT = "fileType";
const _fTM = "finalTextMessage";
const _fU = "fontUrl";
const _fUa = "faviconUrl";
const _g = "groups";
const _gM = "groupMembers";
const _gN = "groupName";
const _gT = "greaterThan";
const _gTOE = "greaterThanOrEquals";
const _gr = "group";
const _h = "history";
const _hA = "hasAccess";
const _hE = "httpError";
const _hQ = "httpQuery";
const _hRC = "hallucinationReductionConfiguration";
const _hRCa = "hallucinationReductionControl";
const _ht = "http";
const _i = "id";
const _iA = "indexArn";
const _iAA = "idcApplicationArn";
const _iAC = "idcAuthConfiguration";
const _iC = "inlineConfigurations";
const _iCAA = "identityCenterApplicationArn";
const _iCIA = "identityCenterInstanceArn";
const _iCn = "invocationCondition";
const _iCns = "instructionCollection";
const _iE = "instructionExample";
const _iEC = "imageExtractionConfiguration";
const _iES = "imageExtractionStatus";
const _iI = "indexId";
const _iIPA = "iamIdentityProviderArn";
const _iPC = "identityProviderConfiguration";
const _iS = "inputStream";
const _iSD = "imageSourceDetails";
const _iSn = "indexStatistics";
const _iT = "identityType";
const _iTB = "indexedTextBytes";
const _iTDC = "indexedTextDocumentCount";
const _iTTIA = "idcTrustedTokenIssuerArn";
const _iTTIC = "idcTrustedTokenIssuerConfiguration";
const _iTd = "idempotencyToken";
const _iUAG = "includedUsersAndGroups";
const _iUC = "inUseConfiguration";
const _id = "identity";
const _in = "indices";
const _it = "items";
const _k = "key";
const _kIC = "kendraIndexConfiguration";
const _kKI = "kmsKeyId";
const _lA = "lambdaArn";
const _lT = "lessThan";
const _lTOE = "lessThanOrEquals";
const _lU = "logoUrl";
const _lUA = "lastUpdatedAt";
const _lUC = "lastUpdateConfiguration";
const _lV = "longValue";
const _m = "message";
const _mB = "mediaBytes";
const _mCA = "messageCopiedAt";
const _mE = "metadataEvent";
const _mEC = "mediaExtractionConfiguration";
const _mG = "memberGroups";
const _mI = "mediaId";
const _mIe = "messageId";
const _mMT = "mediaMimeType";
const _mR = "memberRelation";
const _mRa = "maxResults";
const _mT = "mimeType";
const _mTe = "membershipType";
const _mU = "memberUsers";
const _mUe = "messageUsefulness";
const _mXML = "metadataXML";
const _md = "md5chksum";
const _me = "metrics";
const _mes = "messages";
const _n = "name";
const _nAC = "noAuthConfiguration";
const _nC = "numberConfiguration";
const _nF = "notFilter";
const _nIC = "nativeIndexConfiguration";
const _nS = "nextSubscription";
const _nT = "nextToken";
const _o = "origins";
const _oACCC = "oAuth2ClientCredentialConfiguration";
const _oAF = "orAllFilters";
const _oC = "orchestrationConfiguration";
const _oF = "outputFormat";
const _oIDCC = "openIDConnectConfiguration";
const _oS = "outputStream";
const _oSu = "outputStyle";
const _op = "operator";
const _p = "principals";
const _pA = "pluginArn";
const _pC = "personalizationConfiguration";
const _pCM = "personalizationControlMode";
const _pCl = "pluginConfiguration";
const _pEHC = "preExtractionHookConfiguration";
const _pEHCo = "postExtractionHookConfiguration";
const _pFNS = "payloadFieldNameSeparator";
const _pI = "pluginId";
const _pMI = "parentMessageId";
const _pT = "pluginType";
const _pU = "presignedUrl";
const _pa = "payload";
const _pe = "perspective";
const _pl = "plugins";
const _po = "policy";
const _pr = "principal";
const _qAC = "qAppsConfiguration";
const _qACM = "qAppsControlMode";
const _qSC = "quickSightConfiguration";
const _qT = "queryText";
const _r = "required";
const _rA = "roleArn";
const _rARN = "resourceARN";
const _rAe = "retrieverArn";
const _rC = "responseConfigurations";
const _rCS = "responseConfigurationSummary";
const _rCe = "relevantContent";
const _rCu = "ruleConfiguration";
const _rI = "resourceId";
const _rIe = "retrieverId";
const _rL = "responseLength";
const _rM = "responseMap";
const _rS = "responseScope";
const _rT = "resourceType";
const _rTu = "ruleType";
const _re = "retrievers";
const _rea = "reason";
const _ret = "retriever";
const _ru = "rules";
const _s = "status";
const _sA = "secretArn";
const _sAc = "scoreAttributes";
const _sAe = "secretsArn";
const _sAo = "sourceAttributions";
const _sAou = "sourceAttribution";
const _sAu = "subscriptionArn";
const _sAub = "submittedAt";
const _sBN = "s3BucketName";
const _sC = "scoreConfidence";
const _sCa = "samlConfiguration";
const _sCt = "stringConfiguration";
const _sD = "sourceDetails";
const _sDIM = "sessionDurationInMinutes";
const _sE = "snippetExcerpt";
const _sF = "statusFilter";
const _sGI = "securityGroupIds";
const _sH = "statusHistory";
const _sI = "statementId";
const _sIu = "subscriptionId";
const _sIub = "subnetIds";
const _sLC = "stringListConfiguration";
const _sLV = "stringListValue";
const _sM = "systemMessage";
const _sMI = "systemMessageId";
const _sMO = "systemMessageOverride";
const _sMT = "systemMessageType";
const _sPCM = "samplePromptsControlMode";
const _sPFGM = "s3PathForGroupMembers";
const _sR = "secretsRole";
const _sS = "syncSchedule";
const _sSy = "syncStatus";
const _sT = "startTime";
const _sTM = "startTimeMilliseconds";
const _sU = "serverUrl";
const _sV = "stringValue";
const _s_ = "s3";
const _se = "search";
const _ser = "server";
const _sm = "smithy.ts.sdk.synthetic.com.amazonaws.qbusiness";
const _sn = "snippet";
const _sp = "sparse";
const _st = "statement";
const _sta = "state";
const _str = "streaming";
const _su = "subtitle";
const _sub = "subscriptions";
const _t = "type";
const _tA = "targetAudience";
const _tC = "topicConfigurations";
const _tCTCOU = "topicConfigurationsToCreateOrUpdate";
const _tCTD = "topicConfigurationsToDelete";
const _tDS = "textDocumentStatistics";
const _tE = "textEvent";
const _tK = "tagKeys";
const _tMS = "textMessageSegments";
const _tU = "tokenUrl";
const _ta = "tags";
const _tar = "target";
const _te = "text";
const _ti = "title";
const _tim = "time";
const _to = "tone";
const _u = "users";
const _uA = "updatedAt";
const _uAA = "userAliasesAdded";
const _uAD = "userAliasesDeleted";
const _uATD = "userAliasesToDelete";
const _uATU = "userAliasesToUpdate";
const _uAU = "userAliasesUpdated";
const _uAs = "userAliases";
const _uET = "updatedEarlierThan";
const _uG = "userGroups";
const _uGA = "userGroupAttribute";
const _uI = "userId";
const _uIA = "userIdAttribute";
const _uIs = "userIds";
const _uM = "userMessage";
const _uMI = "userMessageId";
const _un = "units";
const _ur = "url";
const _us = "usefulness";
const _use = "user";
const _v = "value";
const _vC = "vpcConfiguration";
const _vEC = "videoExtractionConfiguration";
const _vES = "videoExtractionStatus";
const _vET = "videoExtractionType";
const _vSD = "videoSourceDetails";
const _ve = "version";
const _wE = "webExperiences";
const _wEA = "webExperienceArn";
const _wEI = "webExperienceId";
const _wM = "welcomeMessage";
const n0 = "com.amazonaws.qbusiness";
var DataAccessorName = [0, n0, _DAN, 8, 0];
var KmsKeyId = [0, n0, _KKI, 8, 0];
var Payload = [0, n0, _P, 8, 0];
var AccessConfiguration = [3, n0, _AC, 0, [_aC, _mR], [() => AccessControls, 0]];
var AccessControl = [3, n0, _ACc, 0, [_p, _mR], [() => Principals, 0]];
var AccessDeniedException = [
    -3,
    n0,
    _ADE,
    {
        [_e]: _c,
        [_hE]: 403,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(AccessDeniedException, AccessDeniedException$1);
var ActionConfiguration = [
    3,
    n0,
    _ACct,
    0,
    [_a, _fC],
    [0, () => ActionFilterConfiguration],
];
var ActionExecution = [
    3,
    n0,
    _AE,
    0,
    [_pI, _pa, _pFNS],
    [0, () => ActionExecutionPayload, 0],
];
var ActionExecutionEvent = [
    3,
    n0,
    _AEE,
    0,
    [_pI, _pa, _pFNS],
    [0, () => ActionExecutionPayload, 0],
];
var ActionExecutionPayloadField = [3, n0, _AEPF, 0, [_v], [15]];
var ActionFilterConfiguration = [3, n0, _AFC, 0, [_dAF], [() => AttributeFilter]];
var ActionReview = [
    3,
    n0,
    _AR,
    0,
    [_pI, _pT, _pa, _pFNS],
    [0, 0, () => ActionReviewPayload, 0],
];
var ActionReviewEvent = [
    3,
    n0,
    _ARE,
    0,
    [_cI, _uMI, _sMI, _pI, _pT, _pa, _pFNS],
    [0, 0, 0, 0, 0, () => ActionReviewPayload, 0],
];
var ActionReviewPayloadField = [
    3,
    n0,
    _ARPF,
    0,
    [_dN, _dO, _dD, _t, _v, _aV, _aF, _aIJS, _r],
    [0, 1, 0, 0, 15, () => ActionReviewPayloadFieldAllowedValues, 0, 15, 2],
];
var ActionReviewPayloadFieldAllowedValue = [3, n0, _ARPFAV, 0, [_v, _dV], [15, 15]];
var ActionSummary = [3, n0, _AS, 0, [_aI, _dN, _iE, _d], [0, 0, 0, 0]];
var Application = [
    3,
    n0,
    _A,
    0,
    [_dN, _aIp, _cA, _uA, _s, _iT, _qSC],
    [0, 0, 4, 4, 0, 0, () => QuickSightConfiguration],
];
var AppliedAttachmentsConfiguration = [3, n0, _AAC, 0, [_aCM], [0]];
var AppliedCreatorModeConfiguration = [3, n0, _ACMC, 0, [_cMC], [0]];
var AppliedOrchestrationConfiguration = [3, n0, _AOC, 0, [_co], [0]];
var AssociatedGroup = [3, n0, _AG, 0, [_n, _t], [0, 0]];
var AssociatedUser = [3, n0, _AU, 0, [_i, _t], [0, 0]];
var AssociatePermissionRequest = [
    3,
    n0,
    _APR,
    0,
    [_aIp, _sI, _ac, _con, _pr],
    [[0, 1], 0, 64 | 0, () => PermissionConditions, 0],
];
var AssociatePermissionResponse = [3, n0, _APRs, 0, [_st], [0]];
var Attachment = [
    3,
    n0,
    _At,
    0,
    [_aIt, _cI, _n, _cF, _fT, _fS, _md, _cA, _s, _e],
    [0, 0, 0, () => CopyFromSource, 0, 1, 0, 4, 0, () => ErrorDetail],
];
var AttachmentInput = [3, n0, _AI, 0, [_da, _n, _cF], [21, 0, () => CopyFromSource]];
var AttachmentInputEvent = [3, n0, _AIE, 0, [_at], [() => AttachmentInput]];
var AttachmentOutput = [
    3,
    n0,
    _AO,
    0,
    [_n, _s, _e, _aIt, _cI],
    [0, 0, () => ErrorDetail, 0, 0],
];
var AttachmentsConfiguration = [3, n0, _ACt, 0, [_aCM], [0]];
var AttributeFilter = [
    3,
    n0,
    _AF,
    0,
    [_aAF, _oAF, _nF, _eT, _cAo, _cAon, _gT, _gTOE, _lT, _lTOE],
    [
        () => AttributeFilters,
        () => AttributeFilters,
        () => AttributeFilter,
        () => DocumentAttribute,
        () => DocumentAttribute,
        () => DocumentAttribute,
        () => DocumentAttribute,
        () => DocumentAttribute,
        () => DocumentAttribute,
        () => DocumentAttribute,
    ],
];
var AudioExtractionConfiguration = [3, n0, _AEC, 0, [_aES], [0]];
var AudioSourceDetails = [3, n0, _ASD, 0, [_mI, _mMT, _sTM, _eTM, _aET], [0, 0, 1, 1, 0]];
var AuthChallengeRequest = [3, n0, _ACR, 0, [_aU], [0]];
var AuthChallengeRequestEvent = [3, n0, _ACRE, 0, [_aU], [0]];
var AuthChallengeResponse = [3, n0, _ACRu, 0, [_rM], [128 | 0]];
var AuthChallengeResponseEvent = [3, n0, _ACREu, 0, [_rM], [128 | 0]];
var AutoSubscriptionConfiguration = [3, n0, _ASC, 0, [_aS, _dST], [0, 0]];
var BasicAuthConfiguration = [3, n0, _BAC, 0, [_sA, _rA], [0, 0]];
var BatchDeleteDocumentRequest = [
    3,
    n0,
    _BDDR,
    0,
    [_aIp, _iI, _do, _dSSI],
    [[0, 1], [0, 1], () => DeleteDocuments, 0],
];
var BatchDeleteDocumentResponse = [3, n0, _BDDRa, 0, [_fD], [() => FailedDocuments]];
var BatchPutDocumentRequest = [
    3,
    n0,
    _BPDR,
    0,
    [_aIp, _iI, _do, _rA, _dSSI],
    [[0, 1], [0, 1], () => Documents, 0, 0],
];
var BatchPutDocumentResponse = [3, n0, _BPDRa, 0, [_fD], [() => FailedDocuments]];
var BlockedPhrasesConfiguration = [3, n0, _BPC, 0, [_bP, _sMO], [64 | 0, 0]];
var BlockedPhrasesConfigurationUpdate = [
    3,
    n0,
    _BPCU,
    0,
    [_bPTCOU, _bPTD, _sMO],
    [64 | 0, 64 | 0, 0],
];
var BrowserExtensionConfiguration = [3, n0, _BEC, 0, [_eBE], [64 | 0]];
var CancelSubscriptionRequest = [
    3,
    n0,
    _CSR,
    0,
    [_aIp, _sIu],
    [
        [0, 1],
        [0, 1],
    ],
];
var CancelSubscriptionResponse = [
    3,
    n0,
    _CSRa,
    0,
    [_sAu, _cS, _nS],
    [0, () => SubscriptionDetails, () => SubscriptionDetails],
];
var ChatInput = [
    3,
    n0,
    _CI,
    0,
    [_aIp, _uI, _uG, _cI, _pMI, _cT, _iS],
    [
        [0, 1],
        [
            0,
            {
                [_hQ]: _uI,
            },
        ],
        [
            64 | 0,
            {
                [_hQ]: _uG,
            },
        ],
        [
            0,
            {
                [_hQ]: _cI,
            },
        ],
        [
            0,
            {
                [_hQ]: _pMI,
            },
        ],
        [
            0,
            {
                [_iTd]: 1,
                [_hQ]: _cT,
            },
        ],
        [() => ChatInputStream, 16],
    ],
];
var ChatOutput = [3, n0, _CO, 0, [_oS], [[() => ChatOutputStream, 16]]];
var ChatResponseConfiguration = [
    3,
    n0,
    _CRC,
    0,
    [_cRCI, _cRCA, _dN, _rCS, _s, _cA, _uA],
    [0, 0, 0, 0, 0, 4, 4],
];
var ChatResponseConfigurationDetail = [
    3,
    n0,
    _CRCD,
    0,
    [_rC, _rCS, _s, _e, _uA],
    [() => ResponseConfigurations, 0, 0, () => ErrorDetail, 4],
];
var ChatSyncInput = [
    3,
    n0,
    _CSI,
    0,
    [_aIp, _uI, _uG, _uM, _att, _aE, _aCR, _cI, _pMI, _aFt, _cM, _cMCh, _cT],
    [
        [0, 1],
        [
            0,
            {
                [_hQ]: _uI,
            },
        ],
        [
            64 | 0,
            {
                [_hQ]: _uG,
            },
        ],
        0,
        () => AttachmentsInput,
        () => ActionExecution,
        () => AuthChallengeResponse,
        0,
        0,
        () => AttributeFilter,
        0,
        () => ChatModeConfiguration,
        [0, 4],
    ],
];
var ChatSyncOutput = [
    3,
    n0,
    _CSO,
    0,
    [_cI, _sM, _sMI, _uMI, _aR, _aCRu, _sAo, _fA],
    [0, 0, 0, 0, () => ActionReview, () => AuthChallengeRequest, [() => SourceAttributions, 0], () => AttachmentsOutput],
];
var CheckDocumentAccessRequest = [
    3,
    n0,
    _CDAR,
    0,
    [_aIp, _iI, _uI, _dI, _dSI],
    [
        [0, 1],
        [0, 1],
        [0, 1],
        [0, 1],
        [
            0,
            {
                [_hQ]: _dSI,
            },
        ],
    ],
];
var CheckDocumentAccessResponse = [
    3,
    n0,
    _CDARh,
    0,
    [_uG, _uAs, _hA, _dA],
    [() => AssociatedGroups, () => AssociatedUsers, 2, () => DocumentAcl],
];
var ConfigurationEvent = [
    3,
    n0,
    _CE,
    0,
    [_cM, _cMCh, _aFt],
    [0, () => ChatModeConfiguration, () => AttributeFilter],
];
var ConflictException = [
    -3,
    n0,
    _CEo,
    {
        [_e]: _c,
        [_hE]: 409,
    },
    [_m, _rI, _rT],
    [0, 0, 0],
];
schema.TypeRegistry.for(n0).registerError(ConflictException, ConflictException$1);
var ContentBlockerRule = [3, n0, _CBR, 0, [_sMO], [0]];
var ContentRetrievalRule = [3, n0, _CRR, 0, [_eDS], [() => EligibleDataSources]];
var Conversation = [3, n0, _C, 0, [_cI, _ti, _sT], [0, 0, 4]];
var ConversationSource = [3, n0, _CS, 0, [_cI, _aIt], [0, 0]];
var CreateAnonymousWebExperienceUrlRequest = [
    3,
    n0,
    _CAWEUR,
    0,
    [_aIp, _wEI, _sDIM],
    [[0, 1], [0, 1], 1],
];
var CreateAnonymousWebExperienceUrlResponse = [3, n0, _CAWEURr, 0, [_aUn], [0]];
var CreateApplicationRequest = [
    3,
    n0,
    _CAR,
    0,
    [_dN, _rA, _iT, _iIPA, _iCIA, _cIFOIDC, _d, _eC, _ta, _cT, _aCt, _qAC, _pC, _qSC],
    [
        0,
        0,
        0,
        0,
        0,
        64 | 0,
        0,
        [() => EncryptionConfiguration, 0],
        () => Tags,
        [0, 4],
        () => AttachmentsConfiguration,
        () => QAppsConfiguration,
        () => PersonalizationConfiguration,
        () => QuickSightConfiguration,
    ],
];
var CreateApplicationResponse = [3, n0, _CARr, 0, [_aIp, _aA], [0, 0]];
var CreateChatResponseConfigurationRequest = [
    3,
    n0,
    _CCRCR,
    0,
    [_aIp, _dN, _cT, _rC, _ta],
    [[0, 1], 0, [0, 4], () => ResponseConfigurations, () => Tags],
];
var CreateChatResponseConfigurationResponse = [3, n0, _CCRCRr, 0, [_cRCI, _cRCA], [0, 0]];
var CreateDataAccessorRequest = [
    3,
    n0,
    _CDARr,
    0,
    [_aIp, _pr, _aCc, _cT, _dN, _aD, _ta],
    [
        [0, 1],
        0,
        () => ActionConfigurationList,
        [0, 4],
        [() => DataAccessorName, 0],
        () => DataAccessorAuthenticationDetail,
        () => Tags,
    ],
];
var CreateDataAccessorResponse = [3, n0, _CDARre, 0, [_dAI, _iAA, _dAA], [0, 0, 0]];
var CreateDataSourceRequest = [
    3,
    n0,
    _CDSR,
    0,
    [_aIp, _iI, _dN, _conf, _vC, _d, _ta, _sS, _rA, _cT, _dEC, _mEC],
    [
        [0, 1],
        [0, 1],
        0,
        15,
        () => DataSourceVpcConfiguration,
        0,
        () => Tags,
        0,
        0,
        [0, 4],
        () => DocumentEnrichmentConfiguration,
        () => MediaExtractionConfiguration,
    ],
];
var CreateDataSourceResponse = [3, n0, _CDSRr, 0, [_dSI, _dSA], [0, 0]];
var CreateIndexRequest = [
    3,
    n0,
    _CIR,
    0,
    [_aIp, _dN, _d, _t, _ta, _cC, _cT],
    [[0, 1], 0, 0, 0, () => Tags, () => IndexCapacityConfiguration, [0, 4]],
];
var CreateIndexResponse = [3, n0, _CIRr, 0, [_iI, _iA], [0, 0]];
var CreatePluginRequest = [
    3,
    n0,
    _CPR,
    0,
    [_aIp, _dN, _t, _aCu, _sU, _cPC, _ta, _cT],
    [[0, 1], 0, 0, () => PluginAuthConfiguration, 0, [() => CustomPluginConfiguration, 0], () => Tags, [0, 4]],
];
var CreatePluginResponse = [3, n0, _CPRr, 0, [_pI, _pA, _bS], [0, 0, 0]];
var CreateRetrieverRequest = [
    3,
    n0,
    _CRRr,
    0,
    [_aIp, _t, _dN, _conf, _rA, _cT, _ta],
    [[0, 1], 0, 0, () => RetrieverConfiguration, 0, [0, 4], () => Tags],
];
var CreateRetrieverResponse = [3, n0, _CRRre, 0, [_rIe, _rAe], [0, 0]];
var CreateSubscriptionRequest = [
    3,
    n0,
    _CSRr,
    0,
    [_aIp, _pr, _t, _cT],
    [[0, 1], () => SubscriptionPrincipal, 0, [0, 4]],
];
var CreateSubscriptionResponse = [
    3,
    n0,
    _CSRre,
    0,
    [_sIu, _sAu, _cS, _nS],
    [0, 0, () => SubscriptionDetails, () => SubscriptionDetails],
];
var CreateUserRequest = [
    3,
    n0,
    _CUR,
    0,
    [_aIp, _uI, _uAs, _cT],
    [[0, 1], 0, () => UserAliases, [0, 4]],
];
var CreateUserResponse = [3, n0, _CURr, 0, [], []];
var CreateWebExperienceRequest = [
    3,
    n0,
    _CWER,
    0,
    [_aIp, _ti, _su, _wM, _sPCM, _o, _rA, _ta, _cT, _iPC, _bEC, _cCu],
    [
        [0, 1],
        0,
        0,
        0,
        0,
        64 | 0,
        0,
        () => Tags,
        [0, 4],
        () => IdentityProviderConfiguration,
        () => BrowserExtensionConfiguration,
        () => CustomizationConfiguration,
    ],
];
var CreateWebExperienceResponse = [3, n0, _CWERr, 0, [_wEI, _wEA], [0, 0]];
var CreatorModeConfiguration = [3, n0, _CMC, 0, [_cMC], [0]];
var CustomizationConfiguration = [3, n0, _CC, 0, [_cCSSU, _lU, _fU, _fUa], [0, 0, 0, 0]];
var CustomPluginConfiguration = [
    3,
    n0,
    _CPC,
    0,
    [_d, _aST, _aSp],
    [0, 0, [() => APISchema, 0]],
];
var DataAccessor = [
    3,
    n0,
    _DA,
    0,
    [_dN, _dAI, _dAA, _iAA, _pr, _aD, _cA, _uA],
    [[() => DataAccessorName, 0], 0, 0, 0, 0, () => DataAccessorAuthenticationDetail, 4, 4],
];
var DataAccessorAuthenticationDetail = [
    3,
    n0,
    _DAAD,
    0,
    [_aT, _aCut, _eI],
    [0, () => DataAccessorAuthenticationConfiguration, 64 | 0],
];
var DataAccessorIdcTrustedTokenIssuerConfiguration = [3, n0, _DAITTIC, 0, [_iTTIA], [0]];
var DataSource = [3, n0, _DS, 0, [_dN, _dSI, _t, _cA, _uA, _s], [0, 0, 0, 4, 4, 0]];
var DataSourceSyncJob = [
    3,
    n0,
    _DSSJ,
    0,
    [_eIx, _sT, _eTn, _s, _e, _dSEC, _me],
    [0, 4, 4, 0, () => ErrorDetail, 0, () => DataSourceSyncJobMetrics],
];
var DataSourceSyncJobMetrics = [
    3,
    n0,
    _DSSJM,
    0,
    [_dAo, _dM, _dDo, _dF, _dS],
    [0, 0, 0, 0, 0],
];
var DataSourceVpcConfiguration = [3, n0, _DSVC, 0, [_sIub, _sGI], [64 | 0, 64 | 0]];
var DateAttributeBoostingConfiguration = [3, n0, _DABC, 0, [_bL, _bDIS], [0, 1]];
var DeleteApplicationRequest = [3, n0, _DAR, 0, [_aIp], [[0, 1]]];
var DeleteApplicationResponse = [3, n0, _DARe, 0, [], []];
var DeleteAttachmentRequest = [
    3,
    n0,
    _DARel,
    0,
    [_aIp, _cI, _aIt, _uI],
    [
        [0, 1],
        [0, 1],
        [0, 1],
        [
            0,
            {
                [_hQ]: _uI,
            },
        ],
    ],
];
var DeleteAttachmentResponse = [3, n0, _DARele, 0, [], []];
var DeleteChatControlsConfigurationRequest = [3, n0, _DCCCR, 0, [_aIp], [[0, 1]]];
var DeleteChatControlsConfigurationResponse = [3, n0, _DCCCRe, 0, [], []];
var DeleteChatResponseConfigurationRequest = [
    3,
    n0,
    _DCRCR,
    0,
    [_aIp, _cRCI],
    [
        [0, 1],
        [0, 1],
    ],
];
var DeleteChatResponseConfigurationResponse = [3, n0, _DCRCRe, 0, [], []];
var DeleteConversationRequest = [
    3,
    n0,
    _DCR,
    0,
    [_cI, _aIp, _uI],
    [
        [0, 1],
        [0, 1],
        [
            0,
            {
                [_hQ]: _uI,
            },
        ],
    ],
];
var DeleteConversationResponse = [3, n0, _DCRe, 0, [], []];
var DeleteDataAccessorRequest = [
    3,
    n0,
    _DDAR,
    0,
    [_aIp, _dAI],
    [
        [0, 1],
        [0, 1],
    ],
];
var DeleteDataAccessorResponse = [3, n0, _DDARe, 0, [], []];
var DeleteDataSourceRequest = [
    3,
    n0,
    _DDSR,
    0,
    [_aIp, _iI, _dSI],
    [
        [0, 1],
        [0, 1],
        [0, 1],
    ],
];
var DeleteDataSourceResponse = [3, n0, _DDSRe, 0, [], []];
var DeleteDocument = [3, n0, _DD, 0, [_dI], [0]];
var DeleteGroupRequest = [
    3,
    n0,
    _DGR,
    0,
    [_aIp, _iI, _gN, _dSI],
    [
        [0, 1],
        [0, 1],
        [0, 1],
        [
            0,
            {
                [_hQ]: _dSI,
            },
        ],
    ],
];
var DeleteGroupResponse = [3, n0, _DGRe, 0, [], []];
var DeleteIndexRequest = [
    3,
    n0,
    _DIR,
    0,
    [_aIp, _iI],
    [
        [0, 1],
        [0, 1],
    ],
];
var DeleteIndexResponse = [3, n0, _DIRe, 0, [], []];
var DeletePluginRequest = [
    3,
    n0,
    _DPR,
    0,
    [_aIp, _pI],
    [
        [0, 1],
        [0, 1],
    ],
];
var DeletePluginResponse = [3, n0, _DPRe, 0, [], []];
var DeleteRetrieverRequest = [
    3,
    n0,
    _DRR,
    0,
    [_aIp, _rIe],
    [
        [0, 1],
        [0, 1],
    ],
];
var DeleteRetrieverResponse = [3, n0, _DRRe, 0, [], []];
var DeleteUserRequest = [
    3,
    n0,
    _DUR,
    0,
    [_aIp, _uI],
    [
        [0, 1],
        [0, 1],
    ],
];
var DeleteUserResponse = [3, n0, _DURe, 0, [], []];
var DeleteWebExperienceRequest = [
    3,
    n0,
    _DWER,
    0,
    [_aIp, _wEI],
    [
        [0, 1],
        [0, 1],
    ],
];
var DeleteWebExperienceResponse = [3, n0, _DWERe, 0, [], []];
var DisassociatePermissionRequest = [
    3,
    n0,
    _DPRi,
    0,
    [_aIp, _sI],
    [
        [0, 1],
        [0, 1],
    ],
];
var DisassociatePermissionResponse = [3, n0, _DPRis, 0, [], []];
var Document = [
    3,
    n0,
    _D,
    0,
    [_i, _attr, _cont, _cTo, _ti, _aCcc, _dEC, _mEC],
    [
        0,
        () => DocumentAttributes,
        () => DocumentContent,
        0,
        0,
        () => AccessConfiguration,
        () => DocumentEnrichmentConfiguration,
        () => MediaExtractionConfiguration,
    ],
];
var DocumentAcl = [
    3,
    n0,
    _DAo,
    0,
    [_al, _dL],
    [() => DocumentAclMembership, () => DocumentAclMembership],
];
var DocumentAclCondition = [
    3,
    n0,
    _DAC,
    0,
    [_mR, _u, _g],
    [0, () => DocumentAclUsers, () => DocumentAclGroups],
];
var DocumentAclGroup = [3, n0, _DAG, 0, [_n, _t], [0, 0]];
var DocumentAclMembership = [
    3,
    n0,
    _DAM,
    0,
    [_mR, _con],
    [0, () => DocumentAclConditions],
];
var DocumentAclUser = [3, n0, _DAU, 0, [_i, _t], [0, 0]];
var DocumentAttribute = [3, n0, _DAoc, 0, [_n, _v], [0, () => DocumentAttributeValue]];
var DocumentAttributeCondition = [
    3,
    n0,
    _DACo,
    0,
    [_k, _op, _v],
    [0, 0, () => DocumentAttributeValue],
];
var DocumentAttributeConfiguration = [3, n0, _DACoc, 0, [_n, _t, _se], [0, 0, 0]];
var DocumentAttributeTarget = [
    3,
    n0,
    _DAT,
    0,
    [_k, _v, _aVO],
    [0, () => DocumentAttributeValue, 0],
];
var DocumentDetails = [
    3,
    n0,
    _DDo,
    0,
    [_dI, _s, _e, _cA, _uA],
    [0, 0, () => ErrorDetail, 4, 4],
];
var DocumentEnrichmentConfiguration = [
    3,
    n0,
    _DEC,
    0,
    [_iC, _pEHC, _pEHCo],
    [() => InlineDocumentEnrichmentConfigurations, () => HookConfiguration, () => HookConfiguration],
];
var EligibleDataSource = [3, n0, _EDS, 0, [_iI, _dSI], [0, 0]];
var EncryptionConfiguration = [3, n0, _EC, 0, [_kKI], [[() => KmsKeyId, 0]]];
var EndOfInputEvent = [3, n0, _EOIE, 0, [], []];
var ErrorDetail = [3, n0, _ED, 0, [_eM, _eCr], [0, 0]];
var ExternalResourceException = [
    -3,
    n0,
    _ERE,
    {
        [_e]: _c,
        [_hE]: 424,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(ExternalResourceException, ExternalResourceException$1);
var FailedAttachmentEvent = [
    3,
    n0,
    _FAE,
    0,
    [_cI, _uMI, _sMI, _at],
    [0, 0, 0, () => AttachmentOutput],
];
var FailedDocument = [3, n0, _FD, 0, [_i, _e, _dSI], [0, () => ErrorDetail, 0]];
var GetApplicationRequest = [3, n0, _GAR, 0, [_aIp], [[0, 1]]];
var GetApplicationResponse = [
    3,
    n0,
    _GARe,
    0,
    [_dN, _aIp, _aA, _iT, _iIPA, _iCAA, _rA, _s, _d, _eC, _cA, _uA, _e, _aCt, _qAC, _pC, _aSC, _cIFOIDC, _qSC],
    [
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        [() => EncryptionConfiguration, 0],
        4,
        4,
        () => ErrorDetail,
        () => AppliedAttachmentsConfiguration,
        () => QAppsConfiguration,
        () => PersonalizationConfiguration,
        () => AutoSubscriptionConfiguration,
        64 | 0,
        () => QuickSightConfiguration,
    ],
];
var GetChatControlsConfigurationRequest = [
    3,
    n0,
    _GCCCR,
    0,
    [_aIp, _mRa, _nT],
    [
        [0, 1],
        [
            1,
            {
                [_hQ]: _mRa,
            },
        ],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
    ],
];
var GetChatControlsConfigurationResponse = [
    3,
    n0,
    _GCCCRe,
    0,
    [_rS, _oC, _bP, _tC, _cMCr, _nT, _hRC],
    [
        0,
        () => AppliedOrchestrationConfiguration,
        () => BlockedPhrasesConfiguration,
        () => TopicConfigurations,
        () => AppliedCreatorModeConfiguration,
        0,
        () => HallucinationReductionConfiguration,
    ],
];
var GetChatResponseConfigurationRequest = [
    3,
    n0,
    _GCRCR,
    0,
    [_aIp, _cRCI],
    [
        [0, 1],
        [0, 1],
    ],
];
var GetChatResponseConfigurationResponse = [
    3,
    n0,
    _GCRCRe,
    0,
    [_cRCI, _cRCA, _dN, _cA, _iUC, _lUC],
    [0, 0, 0, 4, () => ChatResponseConfigurationDetail, () => ChatResponseConfigurationDetail],
];
var GetDataAccessorRequest = [
    3,
    n0,
    _GDAR,
    0,
    [_aIp, _dAI],
    [
        [0, 1],
        [0, 1],
    ],
];
var GetDataAccessorResponse = [
    3,
    n0,
    _GDARe,
    0,
    [_dN, _dAI, _dAA, _aIp, _iAA, _pr, _aCc, _aD, _cA, _uA],
    [
        [() => DataAccessorName, 0],
        0,
        0,
        0,
        0,
        0,
        () => ActionConfigurationList,
        () => DataAccessorAuthenticationDetail,
        4,
        4,
    ],
];
var GetDataSourceRequest = [
    3,
    n0,
    _GDSR,
    0,
    [_aIp, _iI, _dSI],
    [
        [0, 1],
        [0, 1],
        [0, 1],
    ],
];
var GetDataSourceResponse = [
    3,
    n0,
    _GDSRe,
    0,
    [_aIp, _iI, _dSI, _dSA, _dN, _t, _conf, _vC, _cA, _uA, _d, _s, _sS, _rA, _e, _dEC, _mEC],
    [
        0,
        0,
        0,
        0,
        0,
        0,
        15,
        () => DataSourceVpcConfiguration,
        4,
        4,
        0,
        0,
        0,
        0,
        () => ErrorDetail,
        () => DocumentEnrichmentConfiguration,
        () => MediaExtractionConfiguration,
    ],
];
var GetDocumentContentRequest = [
    3,
    n0,
    _GDCR,
    0,
    [_aIp, _iI, _dSI, _dI, _oF],
    [
        [0, 1],
        [0, 1],
        [
            0,
            {
                [_hQ]: _dSI,
            },
        ],
        [0, 1],
        [
            0,
            {
                [_hQ]: _oF,
            },
        ],
    ],
];
var GetDocumentContentResponse = [3, n0, _GDCRe, 0, [_pU, _mT], [0, 0]];
var GetGroupRequest = [
    3,
    n0,
    _GGR,
    0,
    [_aIp, _iI, _gN, _dSI],
    [
        [0, 1],
        [0, 1],
        [0, 1],
        [
            0,
            {
                [_hQ]: _dSI,
            },
        ],
    ],
];
var GetGroupResponse = [
    3,
    n0,
    _GGRe,
    0,
    [_s, _sH],
    [() => GroupStatusDetail, () => GroupStatusDetails],
];
var GetIndexRequest = [
    3,
    n0,
    _GIR,
    0,
    [_aIp, _iI],
    [
        [0, 1],
        [0, 1],
    ],
];
var GetIndexResponse = [
    3,
    n0,
    _GIRe,
    0,
    [_aIp, _iI, _dN, _iA, _s, _t, _d, _cA, _uA, _cC, _dAC, _e, _iSn],
    [
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        4,
        4,
        () => IndexCapacityConfiguration,
        () => DocumentAttributeConfigurations,
        () => ErrorDetail,
        () => IndexStatistics,
    ],
];
var GetMediaRequest = [
    3,
    n0,
    _GMR,
    0,
    [_aIp, _cI, _mIe, _mI],
    [
        [0, 1],
        [0, 1],
        [0, 1],
        [0, 1],
    ],
];
var GetMediaResponse = [3, n0, _GMRe, 0, [_mB, _mMT], [21, 0]];
var GetPluginRequest = [
    3,
    n0,
    _GPR,
    0,
    [_aIp, _pI],
    [
        [0, 1],
        [0, 1],
    ],
];
var GetPluginResponse = [
    3,
    n0,
    _GPRe,
    0,
    [_aIp, _pI, _dN, _t, _sU, _aCu, _cPC, _bS, _pA, _sta, _cA, _uA],
    [0, 0, 0, 0, 0, () => PluginAuthConfiguration, [() => CustomPluginConfiguration, 0], 0, 0, 0, 4, 4],
];
var GetPolicyRequest = [3, n0, _GPRet, 0, [_aIp], [[0, 1]]];
var GetPolicyResponse = [3, n0, _GPReto, 0, [_po], [0]];
var GetRetrieverRequest = [
    3,
    n0,
    _GRR,
    0,
    [_aIp, _rIe],
    [
        [0, 1],
        [0, 1],
    ],
];
var GetRetrieverResponse = [
    3,
    n0,
    _GRRe,
    0,
    [_aIp, _rIe, _rAe, _t, _s, _dN, _conf, _rA, _cA, _uA],
    [0, 0, 0, 0, 0, 0, () => RetrieverConfiguration, 0, 4, 4],
];
var GetUserRequest = [
    3,
    n0,
    _GUR,
    0,
    [_aIp, _uI],
    [
        [0, 1],
        [0, 1],
    ],
];
var GetUserResponse = [3, n0, _GURe, 0, [_uAs], [() => UserAliases]];
var GetWebExperienceRequest = [
    3,
    n0,
    _GWER,
    0,
    [_aIp, _wEI],
    [
        [0, 1],
        [0, 1],
    ],
];
var GetWebExperienceResponse = [
    3,
    n0,
    _GWERe,
    0,
    [_aIp, _wEI, _wEA, _dE, _s, _cA, _uA, _ti, _su, _wM, _sPCM, _o, _rA, _iPC, _aCut, _e, _bEC, _cCu],
    [
        0,
        0,
        0,
        0,
        0,
        4,
        4,
        0,
        0,
        0,
        0,
        64 | 0,
        0,
        () => IdentityProviderConfiguration,
        () => WebExperienceAuthConfiguration,
        () => ErrorDetail,
        () => BrowserExtensionConfiguration,
        () => CustomizationConfiguration,
    ],
];
var GroupMembers = [
    3,
    n0,
    _GM,
    0,
    [_mG, _mU, _sPFGM],
    [() => MemberGroups, () => MemberUsers, () => S3],
];
var GroupStatusDetail = [3, n0, _GSD, 0, [_s, _lUA, _eD], [0, 4, () => ErrorDetail]];
var GroupSummary = [3, n0, _GS, 0, [_gN], [0]];
var HallucinationReductionConfiguration = [3, n0, _HRC, 0, [_hRCa], [0]];
var HookConfiguration = [
    3,
    n0,
    _HC,
    0,
    [_iCn, _lA, _sBN, _rA],
    [() => DocumentAttributeCondition, 0, 0, 0],
];
var IdcAuthConfiguration = [3, n0, _IAC, 0, [_iAA, _rA], [0, 0]];
var ImageExtractionConfiguration = [3, n0, _IEC, 0, [_iES], [0]];
var ImageSourceDetails = [3, n0, _ISD, 0, [_mI, _mMT], [0, 0]];
var Index = [3, n0, _I, 0, [_dN, _iI, _cA, _uA, _s], [0, 0, 4, 4, 0]];
var IndexCapacityConfiguration = [3, n0, _ICC, 0, [_un], [1]];
var IndexStatistics = [3, n0, _IS, 0, [_tDS], [() => TextDocumentStatistics]];
var InlineDocumentEnrichmentConfiguration = [
    3,
    n0,
    _IDEC,
    0,
    [_cond, _tar, _dCO],
    [() => DocumentAttributeCondition, () => DocumentAttributeTarget, 0],
];
var InstructionCollection = [
    3,
    n0,
    _IC,
    0,
    [_rL, _tA, _pe, _oSu, _id, _to, _cIu, _ex],
    [0, 0, 0, 0, 0, 0, 0, 0],
];
var InternalServerException = [
    -3,
    n0,
    _ISE,
    {
        [_e]: _ser,
        [_hE]: 500,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InternalServerException, InternalServerException$1);
var KendraIndexConfiguration = [3, n0, _KIC, 0, [_iI], [0]];
var LicenseNotFoundException = [
    -3,
    n0,
    _LNFE,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(LicenseNotFoundException, LicenseNotFoundException$1);
var ListApplicationsRequest = [
    3,
    n0,
    _LAR,
    0,
    [_nT, _mRa],
    [
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            1,
            {
                [_hQ]: _mRa,
            },
        ],
    ],
];
var ListApplicationsResponse = [3, n0, _LARi, 0, [_nT, _ap], [0, () => Applications]];
var ListAttachmentsRequest = [
    3,
    n0,
    _LARis,
    0,
    [_aIp, _cI, _uI, _nT, _mRa],
    [
        [0, 1],
        [
            0,
            {
                [_hQ]: _cI,
            },
        ],
        [
            0,
            {
                [_hQ]: _uI,
            },
        ],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            1,
            {
                [_hQ]: _mRa,
            },
        ],
    ],
];
var ListAttachmentsResponse = [3, n0, _LARist, 0, [_att, _nT], [() => AttachmentList, 0]];
var ListChatResponseConfigurationsRequest = [
    3,
    n0,
    _LCRCR,
    0,
    [_aIp, _mRa, _nT],
    [
        [0, 1],
        [
            1,
            {
                [_hQ]: _mRa,
            },
        ],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
    ],
];
var ListChatResponseConfigurationsResponse = [
    3,
    n0,
    _LCRCRi,
    0,
    [_cRC, _nT],
    [() => ChatResponseConfigurations, 0],
];
var ListConversationsRequest = [
    3,
    n0,
    _LCR,
    0,
    [_aIp, _uI, _nT, _mRa],
    [
        [0, 1],
        [
            0,
            {
                [_hQ]: _uI,
            },
        ],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            1,
            {
                [_hQ]: _mRa,
            },
        ],
    ],
];
var ListConversationsResponse = [3, n0, _LCRi, 0, [_nT, _conv], [0, () => Conversations]];
var ListDataAccessorsRequest = [
    3,
    n0,
    _LDAR,
    0,
    [_aIp, _nT, _mRa],
    [
        [0, 1],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            1,
            {
                [_hQ]: _mRa,
            },
        ],
    ],
];
var ListDataAccessorsResponse = [
    3,
    n0,
    _LDARi,
    0,
    [_dAa, _nT],
    [[() => DataAccessors, 0], 0],
];
var ListDataSourcesRequest = [
    3,
    n0,
    _LDSR,
    0,
    [_aIp, _iI, _nT, _mRa],
    [
        [0, 1],
        [0, 1],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            1,
            {
                [_hQ]: _mRa,
            },
        ],
    ],
];
var ListDataSourcesResponse = [3, n0, _LDSRi, 0, [_dSa, _nT], [() => DataSources, 0]];
var ListDataSourceSyncJobsRequest = [
    3,
    n0,
    _LDSSJR,
    0,
    [_dSI, _aIp, _iI, _nT, _mRa, _sT, _eTn, _sF],
    [
        [0, 1],
        [0, 1],
        [0, 1],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            1,
            {
                [_hQ]: _mRa,
            },
        ],
        [
            4,
            {
                [_hQ]: _sT,
            },
        ],
        [
            4,
            {
                [_hQ]: _eTn,
            },
        ],
        [
            0,
            {
                [_hQ]: _sSy,
            },
        ],
    ],
];
var ListDataSourceSyncJobsResponse = [
    3,
    n0,
    _LDSSJRi,
    0,
    [_h, _nT],
    [() => DataSourceSyncJobs, 0],
];
var ListDocumentsRequest = [
    3,
    n0,
    _LDR,
    0,
    [_aIp, _iI, _dSIa, _nT, _mRa],
    [
        [0, 1],
        [0, 1],
        [
            64 | 0,
            {
                [_hQ]: _dSIa,
            },
        ],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            1,
            {
                [_hQ]: _mRa,
            },
        ],
    ],
];
var ListDocumentsResponse = [3, n0, _LDRi, 0, [_dDL, _nT], [() => DocumentDetailList, 0]];
var ListGroupsRequest = [
    3,
    n0,
    _LGR,
    0,
    [_aIp, _iI, _uET, _dSI, _nT, _mRa],
    [
        [0, 1],
        [0, 1],
        [
            4,
            {
                [_hQ]: _uET,
            },
        ],
        [
            0,
            {
                [_hQ]: _dSI,
            },
        ],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            1,
            {
                [_hQ]: _mRa,
            },
        ],
    ],
];
var ListGroupsResponse = [3, n0, _LGRi, 0, [_nT, _it], [0, () => GroupSummaryList]];
var ListIndicesRequest = [
    3,
    n0,
    _LIR,
    0,
    [_aIp, _nT, _mRa],
    [
        [0, 1],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            1,
            {
                [_hQ]: _mRa,
            },
        ],
    ],
];
var ListIndicesResponse = [3, n0, _LIRi, 0, [_nT, _in], [0, () => Indices]];
var ListMessagesRequest = [
    3,
    n0,
    _LMR,
    0,
    [_cI, _aIp, _uI, _nT, _mRa],
    [
        [0, 1],
        [0, 1],
        [
            0,
            {
                [_hQ]: _uI,
            },
        ],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            1,
            {
                [_hQ]: _mRa,
            },
        ],
    ],
];
var ListMessagesResponse = [3, n0, _LMRi, 0, [_mes, _nT], [[() => Messages, 0], 0]];
var ListPluginActionsRequest = [
    3,
    n0,
    _LPAR,
    0,
    [_aIp, _pI, _nT, _mRa],
    [
        [0, 1],
        [0, 1],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            1,
            {
                [_hQ]: _mRa,
            },
        ],
    ],
];
var ListPluginActionsResponse = [3, n0, _LPARi, 0, [_nT, _it], [0, () => Actions]];
var ListPluginsRequest = [
    3,
    n0,
    _LPR,
    0,
    [_aIp, _nT, _mRa],
    [
        [0, 1],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            1,
            {
                [_hQ]: _mRa,
            },
        ],
    ],
];
var ListPluginsResponse = [3, n0, _LPRi, 0, [_nT, _pl], [0, () => Plugins]];
var ListPluginTypeActionsRequest = [
    3,
    n0,
    _LPTAR,
    0,
    [_pT, _nT, _mRa],
    [
        [0, 1],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            1,
            {
                [_hQ]: _mRa,
            },
        ],
    ],
];
var ListPluginTypeActionsResponse = [3, n0, _LPTARi, 0, [_nT, _it], [0, () => Actions]];
var ListPluginTypeMetadataRequest = [
    3,
    n0,
    _LPTMR,
    0,
    [_nT, _mRa],
    [
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            1,
            {
                [_hQ]: _mRa,
            },
        ],
    ],
];
var ListPluginTypeMetadataResponse = [
    3,
    n0,
    _LPTMRi,
    0,
    [_nT, _it],
    [0, () => ListPluginTypeMetadataSummaries],
];
var ListRetrieversRequest = [
    3,
    n0,
    _LRR,
    0,
    [_aIp, _nT, _mRa],
    [
        [0, 1],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            1,
            {
                [_hQ]: _mRa,
            },
        ],
    ],
];
var ListRetrieversResponse = [3, n0, _LRRi, 0, [_re, _nT], [() => Retrievers, 0]];
var ListSubscriptionsRequest = [
    3,
    n0,
    _LSR,
    0,
    [_aIp, _nT, _mRa],
    [
        [0, 1],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            1,
            {
                [_hQ]: _mRa,
            },
        ],
    ],
];
var ListSubscriptionsResponse = [3, n0, _LSRi, 0, [_nT, _sub], [0, () => Subscriptions]];
var ListTagsForResourceRequest = [3, n0, _LTFRR, 0, [_rARN], [[0, 1]]];
var ListTagsForResourceResponse = [3, n0, _LTFRRi, 0, [_ta], [() => Tags]];
var ListWebExperiencesRequest = [
    3,
    n0,
    _LWER,
    0,
    [_aIp, _nT, _mRa],
    [
        [0, 1],
        [
            0,
            {
                [_hQ]: _nT,
            },
        ],
        [
            1,
            {
                [_hQ]: _mRa,
            },
        ],
    ],
];
var ListWebExperiencesResponse = [
    3,
    n0,
    _LWERi,
    0,
    [_wE, _nT],
    [() => WebExperiences, 0],
];
var MediaExtractionConfiguration = [
    3,
    n0,
    _MEC,
    0,
    [_iEC, _aEC, _vEC],
    [() => ImageExtractionConfiguration, () => AudioExtractionConfiguration, () => VideoExtractionConfiguration],
];
var MediaTooLargeException = [
    -3,
    n0,
    _MTLE,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(MediaTooLargeException, MediaTooLargeException$1);
var MemberGroup = [3, n0, _MG, 0, [_gN, _t], [0, 0]];
var MemberUser = [3, n0, _MU, 0, [_uI, _t], [0, 0]];
var Message = [
    3,
    n0,
    _M,
    0,
    [_mIe, _b, _tim, _t, _att, _sAou, _aR, _aE],
    [0, 0, 4, 0, () => AttachmentsOutput, [() => SourceAttributions, 0], () => ActionReview, () => ActionExecution],
];
var MessageUsefulnessFeedback = [3, n0, _MUF, 0, [_us, _rea, _com, _sAub], [0, 0, 0, 4]];
var MetadataEvent = [
    3,
    n0,
    _ME,
    0,
    [_cI, _uMI, _sMI, _sAo, _fTM],
    [0, 0, 0, [() => SourceAttributions, 0], 0],
];
var NativeIndexConfiguration = [
    3,
    n0,
    _NIC,
    0,
    [_iI, _ve, _bO],
    [0, 1, () => DocumentAttributeBoostingOverrideMap],
];
var NoAuthConfiguration = [3, n0, _NAC, 0, [], []];
var NumberAttributeBoostingConfiguration = [3, n0, _NABC, 0, [_bL, _bT], [0, 0]];
var OAuth2ClientCredentialConfiguration = [
    3,
    n0,
    _OACCC,
    0,
    [_sA, _rA, _aU, _tU],
    [0, 0, 0, 0],
];
var OpenIDConnectProviderConfiguration = [3, n0, _OIDCPC, 0, [_sAe, _sR], [0, 0]];
var OrchestrationConfiguration = [3, n0, _OC, 0, [_co], [0]];
var PermissionCondition = [3, n0, _PC, 0, [_cO, _cK, _cV], [0, 0, 64 | 0]];
var PersonalizationConfiguration = [3, n0, _PCe, 0, [_pCM], [0]];
var Plugin = [
    3,
    n0,
    _Pl,
    0,
    [_pI, _dN, _t, _sU, _sta, _bS, _cA, _uA],
    [0, 0, 0, 0, 0, 0, 4, 4],
];
var PluginConfiguration = [3, n0, _PCl, 0, [_pI], [0]];
var PluginTypeMetadataSummary = [3, n0, _PTMS, 0, [_t, _ca, _d], [0, 0, 0]];
var PrincipalGroup = [3, n0, _PG, 0, [_n, _acc, _mTe], [0, 0, 0]];
var PrincipalUser = [3, n0, _PU, 0, [_i, _acc, _mTe], [0, 0, 0]];
var PutFeedbackRequest = [
    3,
    n0,
    _PFR,
    0,
    [_aIp, _uI, _cI, _mIe, _mCA, _mUe],
    [
        [0, 1],
        [
            0,
            {
                [_hQ]: _uI,
            },
        ],
        [0, 1],
        [0, 1],
        4,
        () => MessageUsefulnessFeedback,
    ],
];
var PutGroupRequest = [
    3,
    n0,
    _PGR,
    0,
    [_aIp, _iI, _gN, _dSI, _t, _gM, _rA],
    [[0, 1], [0, 1], 0, 0, 0, () => GroupMembers, 0],
];
var PutGroupResponse = [3, n0, _PGRu, 0, [], []];
var QAppsConfiguration = [3, n0, _QAC, 0, [_qACM], [0]];
var QuickSightConfiguration = [3, n0, _QSC, 0, [_cN], [0]];
var RelevantContent = [
    3,
    n0,
    _RC,
    0,
    [_cont, _dI, _dT, _dU, _dAoc, _sAc],
    [0, 0, 0, 0, () => DocumentAttributes, () => ScoreAttributes],
];
var ResourceNotFoundException = [
    -3,
    n0,
    _RNFE,
    {
        [_e]: _c,
        [_hE]: 404,
    },
    [_m, _rI, _rT],
    [0, 0, 0],
];
schema.TypeRegistry.for(n0).registerError(ResourceNotFoundException, ResourceNotFoundException$1);
var ResponseConfiguration = [3, n0, _RCe, 0, [_iCns], [() => InstructionCollection]];
var Retriever = [3, n0, _R, 0, [_aIp, _rIe, _t, _s, _dN], [0, 0, 0, 0, 0]];
var RetrieverContentSource = [3, n0, _RCS, 0, [_rIe], [0]];
var Rule = [
    3,
    n0,
    _Ru,
    0,
    [_iUAG, _eUAG, _rTu, _rCu],
    [() => UsersAndGroups, () => UsersAndGroups, 0, () => RuleConfiguration],
];
var S3 = [3, n0, _S, 0, [_bu, _k], [0, 0]];
var SamlConfiguration = [3, n0, _SC, 0, [_mXML, _rA, _uIA, _uGA], [0, 0, 0, 0]];
var SamlProviderConfiguration = [3, n0, _SPC, 0, [_aUu], [0]];
var ScoreAttributes = [3, n0, _SA, 0, [_sC], [0]];
var SearchRelevantContentRequest = [
    3,
    n0,
    _SRCR,
    0,
    [_aIp, _qT, _cSo, _aFt, _mRa, _nT],
    [[0, 1], 0, () => ContentSource, () => AttributeFilter, 1, 0],
];
var SearchRelevantContentResponse = [
    3,
    n0,
    _SRCRe,
    0,
    [_rCe, _nT],
    [() => RelevantContentList, 0],
];
var ServiceQuotaExceededException = [
    -3,
    n0,
    _SQEE,
    {
        [_e]: _c,
        [_hE]: 402,
    },
    [_m, _rI, _rT],
    [0, 0, 0],
];
schema.TypeRegistry.for(n0).registerError(ServiceQuotaExceededException, ServiceQuotaExceededException$1);
var SnippetExcerpt = [3, n0, _SE, 0, [_te], [0]];
var SourceAttribution = [
    3,
    n0,
    _SAo,
    0,
    [_ti, _sn, _ur, _cNi, _uA, _tMS, _dI, _iI, _dIa],
    [0, 0, 0, 1, 4, () => TextSegmentList, 0, 0, 0],
];
var StartDataSourceSyncJobRequest = [
    3,
    n0,
    _SDSSJR,
    0,
    [_dSI, _aIp, _iI],
    [
        [0, 1],
        [0, 1],
        [0, 1],
    ],
];
var StartDataSourceSyncJobResponse = [3, n0, _SDSSJRt, 0, [_eIx], [0]];
var StopDataSourceSyncJobRequest = [
    3,
    n0,
    _SDSSJRto,
    0,
    [_dSI, _aIp, _iI],
    [
        [0, 1],
        [0, 1],
        [0, 1],
    ],
];
var StopDataSourceSyncJobResponse = [3, n0, _SDSSJRtop, 0, [], []];
var StringAttributeBoostingConfiguration = [3, n0, _SABC, 0, [_bL, _aVB], [0, 128 | 0]];
var StringListAttributeBoostingConfiguration = [3, n0, _SLABC, 0, [_bL], [0]];
var Subscription = [
    3,
    n0,
    _Su,
    0,
    [_sIu, _sAu, _pr, _cS, _nS],
    [0, 0, () => SubscriptionPrincipal, () => SubscriptionDetails, () => SubscriptionDetails],
];
var SubscriptionDetails = [3, n0, _SD, 0, [_t], [0]];
var Tag = [3, n0, _T, 0, [_k, _v], [0, 0]];
var TagResourceRequest = [3, n0, _TRR, 0, [_rARN, _ta], [[0, 1], () => Tags]];
var TagResourceResponse = [3, n0, _TRRa, 0, [], []];
var TextDocumentStatistics = [3, n0, _TDS, 0, [_iTB, _iTDC], [1, 1]];
var TextInputEvent = [3, n0, _TIE, 0, [_uM], [0]];
var TextOutputEvent = [3, n0, _TOE, 0, [_sMT, _cI, _uMI, _sMI, _sM], [0, 0, 0, 0, 0]];
var TextSegment = [
    3,
    n0,
    _TS,
    0,
    [_bOe, _eO, _sE, _mI, _mMT, _sD],
    [1, 1, () => SnippetExcerpt, 0, 0, () => SourceDetails],
];
var ThrottlingException = [
    -3,
    n0,
    _TE,
    {
        [_e]: _c,
        [_hE]: 429,
    },
    [_m],
    [0],
];
schema.TypeRegistry.for(n0).registerError(ThrottlingException, ThrottlingException$1);
var TopicConfiguration = [
    3,
    n0,
    _TC,
    0,
    [_n, _d, _eCM, _ru],
    [0, 0, 64 | 0, () => Rules],
];
var UntagResourceRequest = [
    3,
    n0,
    _URR,
    0,
    [_rARN, _tK],
    [
        [0, 1],
        [
            64 | 0,
            {
                [_hQ]: _tK,
            },
        ],
    ],
];
var UntagResourceResponse = [3, n0, _URRn, 0, [], []];
var UpdateApplicationRequest = [
    3,
    n0,
    _UAR,
    0,
    [_aIp, _iCIA, _dN, _d, _rA, _aCt, _qAC, _pC, _aSC],
    [
        [0, 1],
        0,
        0,
        0,
        0,
        () => AttachmentsConfiguration,
        () => QAppsConfiguration,
        () => PersonalizationConfiguration,
        () => AutoSubscriptionConfiguration,
    ],
];
var UpdateApplicationResponse = [3, n0, _UARp, 0, [], []];
var UpdateChatControlsConfigurationRequest = [
    3,
    n0,
    _UCCCR,
    0,
    [_aIp, _cT, _rS, _oC, _bPCU, _tCTCOU, _tCTD, _cMCr, _hRC],
    [
        [0, 1],
        [0, 4],
        0,
        () => OrchestrationConfiguration,
        () => BlockedPhrasesConfigurationUpdate,
        () => TopicConfigurations,
        () => TopicConfigurations,
        () => CreatorModeConfiguration,
        () => HallucinationReductionConfiguration,
    ],
];
var UpdateChatControlsConfigurationResponse = [3, n0, _UCCCRp, 0, [], []];
var UpdateChatResponseConfigurationRequest = [
    3,
    n0,
    _UCRCR,
    0,
    [_aIp, _cRCI, _dN, _rC, _cT],
    [[0, 1], [0, 1], 0, () => ResponseConfigurations, [0, 4]],
];
var UpdateChatResponseConfigurationResponse = [3, n0, _UCRCRp, 0, [], []];
var UpdateDataAccessorRequest = [
    3,
    n0,
    _UDAR,
    0,
    [_aIp, _dAI, _aCc, _aD, _dN],
    [[0, 1], [0, 1], () => ActionConfigurationList, () => DataAccessorAuthenticationDetail, [() => DataAccessorName, 0]],
];
var UpdateDataAccessorResponse = [3, n0, _UDARp, 0, [], []];
var UpdateDataSourceRequest = [
    3,
    n0,
    _UDSR,
    0,
    [_aIp, _iI, _dSI, _dN, _conf, _vC, _d, _sS, _rA, _dEC, _mEC],
    [
        [0, 1],
        [0, 1],
        [0, 1],
        0,
        15,
        () => DataSourceVpcConfiguration,
        0,
        0,
        0,
        () => DocumentEnrichmentConfiguration,
        () => MediaExtractionConfiguration,
    ],
];
var UpdateDataSourceResponse = [3, n0, _UDSRp, 0, [], []];
var UpdateIndexRequest = [
    3,
    n0,
    _UIR,
    0,
    [_aIp, _iI, _dN, _d, _cC, _dAC],
    [[0, 1], [0, 1], 0, 0, () => IndexCapacityConfiguration, () => DocumentAttributeConfigurations],
];
var UpdateIndexResponse = [3, n0, _UIRp, 0, [], []];
var UpdatePluginRequest = [
    3,
    n0,
    _UPR,
    0,
    [_aIp, _pI, _dN, _sta, _sU, _cPC, _aCu],
    [[0, 1], [0, 1], 0, 0, 0, [() => CustomPluginConfiguration, 0], () => PluginAuthConfiguration],
];
var UpdatePluginResponse = [3, n0, _UPRp, 0, [], []];
var UpdateRetrieverRequest = [
    3,
    n0,
    _URRp,
    0,
    [_aIp, _rIe, _conf, _dN, _rA],
    [[0, 1], [0, 1], () => RetrieverConfiguration, 0, 0],
];
var UpdateRetrieverResponse = [3, n0, _URRpd, 0, [], []];
var UpdateSubscriptionRequest = [3, n0, _USR, 0, [_aIp, _sIu, _t], [[0, 1], [0, 1], 0]];
var UpdateSubscriptionResponse = [
    3,
    n0,
    _USRp,
    0,
    [_sAu, _cS, _nS],
    [0, () => SubscriptionDetails, () => SubscriptionDetails],
];
var UpdateUserRequest = [
    3,
    n0,
    _UUR,
    0,
    [_aIp, _uI, _uATU, _uATD],
    [[0, 1], [0, 1], () => UserAliases, () => UserAliases],
];
var UpdateUserResponse = [
    3,
    n0,
    _UURp,
    0,
    [_uAA, _uAU, _uAD],
    [() => UserAliases, () => UserAliases, () => UserAliases],
];
var UpdateWebExperienceRequest = [
    3,
    n0,
    _UWER,
    0,
    [_aIp, _wEI, _rA, _aCut, _ti, _su, _wM, _sPCM, _iPC, _o, _bEC, _cCu],
    [
        [0, 1],
        [0, 1],
        0,
        () => WebExperienceAuthConfiguration,
        0,
        0,
        0,
        0,
        () => IdentityProviderConfiguration,
        64 | 0,
        () => BrowserExtensionConfiguration,
        () => CustomizationConfiguration,
    ],
];
var UpdateWebExperienceResponse = [3, n0, _UWERp, 0, [], []];
var UserAlias = [3, n0, _UA, 0, [_iI, _dSI, _uI], [0, 0, 0]];
var UsersAndGroups = [3, n0, _UAG, 0, [_uIs, _uG], [64 | 0, 64 | 0]];
var ValidationException = [
    -3,
    n0,
    _VE,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_m, _rea, _f],
    [0, 0, () => ValidationExceptionFields],
];
schema.TypeRegistry.for(n0).registerError(ValidationException, ValidationException$1);
var ValidationExceptionField = [3, n0, _VEF, 0, [_n, _m], [0, 0]];
var VideoExtractionConfiguration = [3, n0, _VEC, 0, [_vES], [0]];
var VideoSourceDetails = [3, n0, _VSD, 0, [_mI, _mMT, _sTM, _eTM, _vET], [0, 0, 1, 1, 0]];
var WebExperience = [3, n0, _WE, 0, [_wEI, _cA, _uA, _dE, _s], [0, 4, 4, 0, 0]];
var __Unit = "unit";
var QBusinessServiceException = [-3, _sm, "QBusinessServiceException", 0, [], []];
schema.TypeRegistry.for(_sm).registerError(QBusinessServiceException, QBusinessServiceException$1);
var AccessControls = [1, n0, _ACcc, 0, () => AccessControl];
var ActionConfigurationList = [1, n0, _ACL, 0, () => ActionConfiguration];
var ActionReviewPayloadFieldAllowedValues = [
    1,
    n0,
    _ARPFAVc,
    0,
    () => ActionReviewPayloadFieldAllowedValue,
];
var Actions = [1, n0, _Ac, 0, () => ActionSummary];
var Applications = [1, n0, _Ap, 0, () => Application];
var AssociatedGroups = [1, n0, _AGs, 0, () => AssociatedGroup];
var AssociatedUsers = [1, n0, _AUs, 0, () => AssociatedUser];
var AttachmentList = [1, n0, _AL, 0, () => Attachment];
var AttachmentsInput = [1, n0, _AIt, 0, () => AttachmentInput];
var AttachmentsOutput = [1, n0, _AOt, 0, () => AttachmentOutput];
var AttributeFilters = [1, n0, _AFt, 0, () => AttributeFilter];
var ChatResponseConfigurations = [1, n0, _CRCh, 0, () => ChatResponseConfiguration];
var Conversations = [1, n0, _Co, 0, () => Conversation];
var DataAccessors = [1, n0, _DAa, 0, [() => DataAccessor, 0]];
var DataSources = [1, n0, _DSa, 0, () => DataSource];
var DataSourceSyncJobs = [1, n0, _DSSJa, 0, () => DataSourceSyncJob];
var DeleteDocuments = [1, n0, _DDe, 0, () => DeleteDocument];
var DocumentAclConditions = [1, n0, _DACocu, 0, () => DocumentAclCondition];
var DocumentAclGroups = [1, n0, _DAGo, 0, () => DocumentAclGroup];
var DocumentAclUsers = [1, n0, _DAUo, 0, () => DocumentAclUser];
var DocumentAttributeConfigurations = [
    1,
    n0,
    _DACocum,
    0,
    () => DocumentAttributeConfiguration,
];
var DocumentAttributes = [1, n0, _DAocu, 0, () => DocumentAttribute];
var DocumentDetailList = [1, n0, _DDL, 0, () => DocumentDetails];
var Documents = [1, n0, _Do, 0, () => Document];
var EligibleDataSources = [1, n0, _EDSl, 0, () => EligibleDataSource];
var FailedDocuments = [1, n0, _FDa, 0, () => FailedDocument];
var GroupStatusDetails = [1, n0, _GSDr, 0, () => GroupStatusDetail];
var GroupSummaryList = [1, n0, _GSL, 0, () => GroupSummary];
var Indices = [1, n0, _In, 0, () => Index];
var InlineDocumentEnrichmentConfigurations = [
    1,
    n0,
    _IDECn,
    0,
    () => InlineDocumentEnrichmentConfiguration,
];
var ListPluginTypeMetadataSummaries = [1, n0, _LPTMS, 0, () => PluginTypeMetadataSummary];
var MemberGroups = [1, n0, _MGe, 0, () => MemberGroup];
var MemberUsers = [1, n0, _MUe, 0, () => MemberUser];
var Messages = [1, n0, _Me, 0, [() => Message, 0]];
var PermissionConditions = [1, n0, _PCer, 0, () => PermissionCondition];
var Plugins = [1, n0, _Plu, 0, () => Plugin];
var Principals = [1, n0, _Pr, 0, () => Principal];
var RelevantContentList = [1, n0, _RCL, 0, () => RelevantContent];
var Retrievers = [1, n0, _Re, 0, () => Retriever];
var Rules = [1, n0, _Rul, 0, () => Rule];
var SourceAttributions = [
    1,
    n0,
    _SAou,
    {
        [_sp]: 1,
    },
    () => SourceAttribution,
];
var Subscriptions = [1, n0, _Sub, 0, () => Subscription];
var Tags = [1, n0, _Ta, 0, () => Tag];
var TextSegmentList = [1, n0, _TSL, 0, () => TextSegment];
var TopicConfigurations = [1, n0, _TCo, 0, () => TopicConfiguration];
var UserAliases = [1, n0, _UAs, 0, () => UserAlias];
var ValidationExceptionFields = [1, n0, _VEFa, 0, () => ValidationExceptionField];
var WebExperiences = [1, n0, _WEe, 0, () => WebExperience];
var ActionExecutionPayload = [2, n0, _AEP, 0, 0, () => ActionExecutionPayloadField];
var ActionReviewPayload = [2, n0, _ARP, 0, 0, () => ActionReviewPayloadField];
var DocumentAttributeBoostingOverrideMap = [
    2,
    n0,
    _DABOM,
    0,
    0,
    () => DocumentAttributeBoostingConfiguration,
];
var ResponseConfigurations = [2, n0, _RCes, 0, 0, () => ResponseConfiguration];
var APISchema = [3, n0, _APIS, 0, [_pa, _s_], [[() => Payload, 0], () => S3]];
var ChatInputStream = [
    3,
    n0,
    _CIS,
    {
        [_str]: 1,
    },
    [_cE, _tE, _aEt, _aEE, _eOIE, _aCRE],
    [
        () => ConfigurationEvent,
        () => TextInputEvent,
        () => AttachmentInputEvent,
        () => ActionExecutionEvent,
        () => EndOfInputEvent,
        () => AuthChallengeResponseEvent,
    ],
];
var ChatModeConfiguration = [3, n0, _CMCh, 0, [_pCl], [() => PluginConfiguration]];
var ChatOutputStream = [
    3,
    n0,
    _COS,
    {
        [_str]: 1,
    },
    [_tE, _mE, _aRE, _fAE, _aCREu],
    [
        () => TextOutputEvent,
        [() => MetadataEvent, 0],
        () => ActionReviewEvent,
        () => FailedAttachmentEvent,
        () => AuthChallengeRequestEvent,
    ],
];
var ContentSource = [3, n0, _CSo, 0, [_ret], [() => RetrieverContentSource]];
var CopyFromSource = [3, n0, _CFS, 0, [_conve], [() => ConversationSource]];
var DataAccessorAuthenticationConfiguration = [
    3,
    n0,
    _DAAC,
    0,
    [_iTTIC],
    [() => DataAccessorIdcTrustedTokenIssuerConfiguration],
];
var DocumentAttributeBoostingConfiguration = [
    3,
    n0,
    _DABCo,
    0,
    [_nC, _sCt, _dC, _sLC],
    [
        () => NumberAttributeBoostingConfiguration,
        () => StringAttributeBoostingConfiguration,
        () => DateAttributeBoostingConfiguration,
        () => StringListAttributeBoostingConfiguration,
    ],
];
var DocumentAttributeValue = [3, n0, _DAV, 0, [_sV, _sLV, _lV, _dVa], [0, 64 | 0, 1, 4]];
var DocumentContent = [3, n0, _DC, 0, [_bl, _s_], [21, () => S3]];
var IdentityProviderConfiguration = [
    3,
    n0,
    _IPC,
    0,
    [_sCa, _oIDCC],
    [() => SamlProviderConfiguration, () => OpenIDConnectProviderConfiguration],
];
var PluginAuthConfiguration = [
    3,
    n0,
    _PAC,
    0,
    [_bAC, _oACCC, _nAC, _iAC],
    [
        () => BasicAuthConfiguration,
        () => OAuth2ClientCredentialConfiguration,
        () => NoAuthConfiguration,
        () => IdcAuthConfiguration,
    ],
];
var Principal = [
    3,
    n0,
    _Pri,
    0,
    [_use, _gr],
    [() => PrincipalUser, () => PrincipalGroup],
];
var RetrieverConfiguration = [
    3,
    n0,
    _RCet,
    0,
    [_nIC, _kIC],
    [() => NativeIndexConfiguration, () => KendraIndexConfiguration],
];
var RuleConfiguration = [
    3,
    n0,
    _RCu,
    0,
    [_cBR, _cRR],
    [() => ContentBlockerRule, () => ContentRetrievalRule],
];
var SourceDetails = [
    3,
    n0,
    _SDo,
    0,
    [_iSD, _aSD, _vSD],
    [() => ImageSourceDetails, () => AudioSourceDetails, () => VideoSourceDetails],
];
var SubscriptionPrincipal = [3, n0, _SP, 0, [_use, _gr], [0, 0]];
var WebExperienceAuthConfiguration = [3, n0, _WEAC, 0, [_sCa], [() => SamlConfiguration]];
var AssociatePermission = [
    9,
    n0,
    _AP,
    {
        [_ht]: ["POST", "/applications/{applicationId}/policy", 200],
    },
    () => AssociatePermissionRequest,
    () => AssociatePermissionResponse,
];
var BatchDeleteDocument = [
    9,
    n0,
    _BDD,
    {
        [_ht]: ["POST", "/applications/{applicationId}/indices/{indexId}/documents/delete", 200],
    },
    () => BatchDeleteDocumentRequest,
    () => BatchDeleteDocumentResponse,
];
var BatchPutDocument = [
    9,
    n0,
    _BPD,
    {
        [_ht]: ["POST", "/applications/{applicationId}/indices/{indexId}/documents", 200],
    },
    () => BatchPutDocumentRequest,
    () => BatchPutDocumentResponse,
];
var CancelSubscription = [
    9,
    n0,
    _CSa,
    {
        [_ht]: ["DELETE", "/applications/{applicationId}/subscriptions/{subscriptionId}", 200],
    },
    () => CancelSubscriptionRequest,
    () => CancelSubscriptionResponse,
];
var Chat = [
    9,
    n0,
    _Ch,
    {
        [_ht]: ["POST", "/applications/{applicationId}/conversations", 200],
    },
    () => ChatInput,
    () => ChatOutput,
];
var ChatSync = [
    9,
    n0,
    _CSh,
    {
        [_ht]: ["POST", "/applications/{applicationId}/conversations?sync", 200],
    },
    () => ChatSyncInput,
    () => ChatSyncOutput,
];
var CheckDocumentAccess = [
    9,
    n0,
    _CDA,
    {
        [_ht]: [
            "GET",
            "/applications/{applicationId}/index/{indexId}/users/{userId}/documents/{documentId}/check-document-access",
            200,
        ],
    },
    () => CheckDocumentAccessRequest,
    () => CheckDocumentAccessResponse,
];
var CreateAnonymousWebExperienceUrl = [
    9,
    n0,
    _CAWEU,
    {
        [_ht]: ["POST", "/applications/{applicationId}/experiences/{webExperienceId}/anonymous-url", 200],
    },
    () => CreateAnonymousWebExperienceUrlRequest,
    () => CreateAnonymousWebExperienceUrlResponse,
];
var CreateApplication = [
    9,
    n0,
    _CA,
    {
        [_ht]: ["POST", "/applications", 200],
    },
    () => CreateApplicationRequest,
    () => CreateApplicationResponse,
];
var CreateChatResponseConfiguration = [
    9,
    n0,
    _CCRC,
    {
        [_ht]: ["POST", "/applications/{applicationId}/chatresponseconfigurations", 200],
    },
    () => CreateChatResponseConfigurationRequest,
    () => CreateChatResponseConfigurationResponse,
];
var CreateDataAccessor = [
    9,
    n0,
    _CDAr,
    {
        [_ht]: ["POST", "/applications/{applicationId}/dataaccessors", 200],
    },
    () => CreateDataAccessorRequest,
    () => CreateDataAccessorResponse,
];
var CreateDataSource = [
    9,
    n0,
    _CDS,
    {
        [_ht]: ["POST", "/applications/{applicationId}/indices/{indexId}/datasources", 200],
    },
    () => CreateDataSourceRequest,
    () => CreateDataSourceResponse,
];
var CreateIndex = [
    9,
    n0,
    _CIr,
    {
        [_ht]: ["POST", "/applications/{applicationId}/indices", 200],
    },
    () => CreateIndexRequest,
    () => CreateIndexResponse,
];
var CreatePlugin = [
    9,
    n0,
    _CP,
    {
        [_ht]: ["POST", "/applications/{applicationId}/plugins", 200],
    },
    () => CreatePluginRequest,
    () => CreatePluginResponse,
];
var CreateRetriever = [
    9,
    n0,
    _CR,
    {
        [_ht]: ["POST", "/applications/{applicationId}/retrievers", 200],
    },
    () => CreateRetrieverRequest,
    () => CreateRetrieverResponse,
];
var CreateSubscription = [
    9,
    n0,
    _CSr,
    {
        [_ht]: ["POST", "/applications/{applicationId}/subscriptions", 200],
    },
    () => CreateSubscriptionRequest,
    () => CreateSubscriptionResponse,
];
var CreateUser = [
    9,
    n0,
    _CU,
    {
        [_ht]: ["POST", "/applications/{applicationId}/users", 200],
    },
    () => CreateUserRequest,
    () => CreateUserResponse,
];
var CreateWebExperience = [
    9,
    n0,
    _CWE,
    {
        [_ht]: ["POST", "/applications/{applicationId}/experiences", 200],
    },
    () => CreateWebExperienceRequest,
    () => CreateWebExperienceResponse,
];
var DeleteApplication = [
    9,
    n0,
    _DAe,
    {
        [_ht]: ["DELETE", "/applications/{applicationId}", 200],
    },
    () => DeleteApplicationRequest,
    () => DeleteApplicationResponse,
];
var DeleteAttachment = [
    9,
    n0,
    _DAel,
    {
        [_ht]: ["DELETE", "/applications/{applicationId}/conversations/{conversationId}/attachments/{attachmentId}", 200],
    },
    () => DeleteAttachmentRequest,
    () => DeleteAttachmentResponse,
];
var DeleteChatControlsConfiguration = [
    9,
    n0,
    _DCCC,
    {
        [_ht]: ["DELETE", "/applications/{applicationId}/chatcontrols", 200],
    },
    () => DeleteChatControlsConfigurationRequest,
    () => DeleteChatControlsConfigurationResponse,
];
var DeleteChatResponseConfiguration = [
    9,
    n0,
    _DCRC,
    {
        [_ht]: ["DELETE", "/applications/{applicationId}/chatresponseconfigurations/{chatResponseConfigurationId}", 200],
    },
    () => DeleteChatResponseConfigurationRequest,
    () => DeleteChatResponseConfigurationResponse,
];
var DeleteConversation = [
    9,
    n0,
    _DCe,
    {
        [_ht]: ["DELETE", "/applications/{applicationId}/conversations/{conversationId}", 200],
    },
    () => DeleteConversationRequest,
    () => DeleteConversationResponse,
];
var DeleteDataAccessor = [
    9,
    n0,
    _DDA,
    {
        [_ht]: ["DELETE", "/applications/{applicationId}/dataaccessors/{dataAccessorId}", 200],
    },
    () => DeleteDataAccessorRequest,
    () => DeleteDataAccessorResponse,
];
var DeleteDataSource = [
    9,
    n0,
    _DDS,
    {
        [_ht]: ["DELETE", "/applications/{applicationId}/indices/{indexId}/datasources/{dataSourceId}", 200],
    },
    () => DeleteDataSourceRequest,
    () => DeleteDataSourceResponse,
];
var DeleteGroup = [
    9,
    n0,
    _DG,
    {
        [_ht]: ["DELETE", "/applications/{applicationId}/indices/{indexId}/groups/{groupName}", 200],
    },
    () => DeleteGroupRequest,
    () => DeleteGroupResponse,
];
var DeleteIndex = [
    9,
    n0,
    _DI,
    {
        [_ht]: ["DELETE", "/applications/{applicationId}/indices/{indexId}", 200],
    },
    () => DeleteIndexRequest,
    () => DeleteIndexResponse,
];
var DeletePlugin = [
    9,
    n0,
    _DP,
    {
        [_ht]: ["DELETE", "/applications/{applicationId}/plugins/{pluginId}", 200],
    },
    () => DeletePluginRequest,
    () => DeletePluginResponse,
];
var DeleteRetriever = [
    9,
    n0,
    _DR,
    {
        [_ht]: ["DELETE", "/applications/{applicationId}/retrievers/{retrieverId}", 200],
    },
    () => DeleteRetrieverRequest,
    () => DeleteRetrieverResponse,
];
var DeleteUser = [
    9,
    n0,
    _DU,
    {
        [_ht]: ["DELETE", "/applications/{applicationId}/users/{userId}", 200],
    },
    () => DeleteUserRequest,
    () => DeleteUserResponse,
];
var DeleteWebExperience = [
    9,
    n0,
    _DWE,
    {
        [_ht]: ["DELETE", "/applications/{applicationId}/experiences/{webExperienceId}", 200],
    },
    () => DeleteWebExperienceRequest,
    () => DeleteWebExperienceResponse,
];
var DisassociatePermission = [
    9,
    n0,
    _DPi,
    {
        [_ht]: ["DELETE", "/applications/{applicationId}/policy/{statementId}", 200],
    },
    () => DisassociatePermissionRequest,
    () => DisassociatePermissionResponse,
];
var GetApplication = [
    9,
    n0,
    _GA,
    {
        [_ht]: ["GET", "/applications/{applicationId}", 200],
    },
    () => GetApplicationRequest,
    () => GetApplicationResponse,
];
var GetChatControlsConfiguration = [
    9,
    n0,
    _GCCC,
    {
        [_ht]: ["GET", "/applications/{applicationId}/chatcontrols", 200],
    },
    () => GetChatControlsConfigurationRequest,
    () => GetChatControlsConfigurationResponse,
];
var GetChatResponseConfiguration = [
    9,
    n0,
    _GCRC,
    {
        [_ht]: ["GET", "/applications/{applicationId}/chatresponseconfigurations/{chatResponseConfigurationId}", 200],
    },
    () => GetChatResponseConfigurationRequest,
    () => GetChatResponseConfigurationResponse,
];
var GetDataAccessor = [
    9,
    n0,
    _GDA,
    {
        [_ht]: ["GET", "/applications/{applicationId}/dataaccessors/{dataAccessorId}", 200],
    },
    () => GetDataAccessorRequest,
    () => GetDataAccessorResponse,
];
var GetDataSource = [
    9,
    n0,
    _GDS,
    {
        [_ht]: ["GET", "/applications/{applicationId}/indices/{indexId}/datasources/{dataSourceId}", 200],
    },
    () => GetDataSourceRequest,
    () => GetDataSourceResponse,
];
var GetDocumentContent = [
    9,
    n0,
    _GDC,
    {
        [_ht]: ["GET", "/applications/{applicationId}/index/{indexId}/documents/{documentId}/content", 200],
    },
    () => GetDocumentContentRequest,
    () => GetDocumentContentResponse,
];
var GetGroup = [
    9,
    n0,
    _GG,
    {
        [_ht]: ["GET", "/applications/{applicationId}/indices/{indexId}/groups/{groupName}", 200],
    },
    () => GetGroupRequest,
    () => GetGroupResponse,
];
var GetIndex = [
    9,
    n0,
    _GI,
    {
        [_ht]: ["GET", "/applications/{applicationId}/indices/{indexId}", 200],
    },
    () => GetIndexRequest,
    () => GetIndexResponse,
];
var GetMedia = [
    9,
    n0,
    _GMe,
    {
        [_ht]: [
            "GET",
            "/applications/{applicationId}/conversations/{conversationId}/messages/{messageId}/media/{mediaId}",
            200,
        ],
    },
    () => GetMediaRequest,
    () => GetMediaResponse,
];
var GetPlugin = [
    9,
    n0,
    _GP,
    {
        [_ht]: ["GET", "/applications/{applicationId}/plugins/{pluginId}", 200],
    },
    () => GetPluginRequest,
    () => GetPluginResponse,
];
var GetPolicy = [
    9,
    n0,
    _GPe,
    {
        [_ht]: ["GET", "/applications/{applicationId}/policy", 200],
    },
    () => GetPolicyRequest,
    () => GetPolicyResponse,
];
var GetRetriever = [
    9,
    n0,
    _GR,
    {
        [_ht]: ["GET", "/applications/{applicationId}/retrievers/{retrieverId}", 200],
    },
    () => GetRetrieverRequest,
    () => GetRetrieverResponse,
];
var GetUser = [
    9,
    n0,
    _GU,
    {
        [_ht]: ["GET", "/applications/{applicationId}/users/{userId}", 200],
    },
    () => GetUserRequest,
    () => GetUserResponse,
];
var GetWebExperience = [
    9,
    n0,
    _GWE,
    {
        [_ht]: ["GET", "/applications/{applicationId}/experiences/{webExperienceId}", 200],
    },
    () => GetWebExperienceRequest,
    () => GetWebExperienceResponse,
];
var ListApplications = [
    9,
    n0,
    _LA,
    {
        [_ht]: ["GET", "/applications", 200],
    },
    () => ListApplicationsRequest,
    () => ListApplicationsResponse,
];
var ListAttachments = [
    9,
    n0,
    _LAi,
    {
        [_ht]: ["GET", "/applications/{applicationId}/attachments", 200],
    },
    () => ListAttachmentsRequest,
    () => ListAttachmentsResponse,
];
var ListChatResponseConfigurations = [
    9,
    n0,
    _LCRC,
    {
        [_ht]: ["GET", "/applications/{applicationId}/chatresponseconfigurations", 200],
    },
    () => ListChatResponseConfigurationsRequest,
    () => ListChatResponseConfigurationsResponse,
];
var ListConversations = [
    9,
    n0,
    _LC,
    {
        [_ht]: ["GET", "/applications/{applicationId}/conversations", 200],
    },
    () => ListConversationsRequest,
    () => ListConversationsResponse,
];
var ListDataAccessors = [
    9,
    n0,
    _LDA,
    {
        [_ht]: ["GET", "/applications/{applicationId}/dataaccessors", 200],
    },
    () => ListDataAccessorsRequest,
    () => ListDataAccessorsResponse,
];
var ListDataSources = [
    9,
    n0,
    _LDS,
    {
        [_ht]: ["GET", "/applications/{applicationId}/indices/{indexId}/datasources", 200],
    },
    () => ListDataSourcesRequest,
    () => ListDataSourcesResponse,
];
var ListDataSourceSyncJobs = [
    9,
    n0,
    _LDSSJ,
    {
        [_ht]: ["GET", "/applications/{applicationId}/indices/{indexId}/datasources/{dataSourceId}/syncjobs", 200],
    },
    () => ListDataSourceSyncJobsRequest,
    () => ListDataSourceSyncJobsResponse,
];
var ListDocuments = [
    9,
    n0,
    _LD,
    {
        [_ht]: ["GET", "/applications/{applicationId}/index/{indexId}/documents", 200],
    },
    () => ListDocumentsRequest,
    () => ListDocumentsResponse,
];
var ListGroups = [
    9,
    n0,
    _LG,
    {
        [_ht]: ["GET", "/applications/{applicationId}/indices/{indexId}/groups", 200],
    },
    () => ListGroupsRequest,
    () => ListGroupsResponse,
];
var ListIndices = [
    9,
    n0,
    _LI,
    {
        [_ht]: ["GET", "/applications/{applicationId}/indices", 200],
    },
    () => ListIndicesRequest,
    () => ListIndicesResponse,
];
var ListMessages = [
    9,
    n0,
    _LM,
    {
        [_ht]: ["GET", "/applications/{applicationId}/conversations/{conversationId}", 200],
    },
    () => ListMessagesRequest,
    () => ListMessagesResponse,
];
var ListPluginActions = [
    9,
    n0,
    _LPA,
    {
        [_ht]: ["GET", "/applications/{applicationId}/plugins/{pluginId}/actions", 200],
    },
    () => ListPluginActionsRequest,
    () => ListPluginActionsResponse,
];
var ListPlugins = [
    9,
    n0,
    _LP,
    {
        [_ht]: ["GET", "/applications/{applicationId}/plugins", 200],
    },
    () => ListPluginsRequest,
    () => ListPluginsResponse,
];
var ListPluginTypeActions = [
    9,
    n0,
    _LPTA,
    {
        [_ht]: ["GET", "/pluginTypes/{pluginType}/actions", 200],
    },
    () => ListPluginTypeActionsRequest,
    () => ListPluginTypeActionsResponse,
];
var ListPluginTypeMetadata = [
    9,
    n0,
    _LPTM,
    {
        [_ht]: ["GET", "/pluginTypeMetadata", 200],
    },
    () => ListPluginTypeMetadataRequest,
    () => ListPluginTypeMetadataResponse,
];
var ListRetrievers = [
    9,
    n0,
    _LR,
    {
        [_ht]: ["GET", "/applications/{applicationId}/retrievers", 200],
    },
    () => ListRetrieversRequest,
    () => ListRetrieversResponse,
];
var ListSubscriptions = [
    9,
    n0,
    _LS,
    {
        [_ht]: ["GET", "/applications/{applicationId}/subscriptions", 200],
    },
    () => ListSubscriptionsRequest,
    () => ListSubscriptionsResponse,
];
var ListTagsForResource = [
    9,
    n0,
    _LTFR,
    {
        [_ht]: ["GET", "/v1/tags/{resourceARN}", 200],
    },
    () => ListTagsForResourceRequest,
    () => ListTagsForResourceResponse,
];
var ListWebExperiences = [
    9,
    n0,
    _LWE,
    {
        [_ht]: ["GET", "/applications/{applicationId}/experiences", 200],
    },
    () => ListWebExperiencesRequest,
    () => ListWebExperiencesResponse,
];
var PutFeedback = [
    9,
    n0,
    _PF,
    {
        [_ht]: ["POST", "/applications/{applicationId}/conversations/{conversationId}/messages/{messageId}/feedback", 200],
    },
    () => PutFeedbackRequest,
    () => __Unit,
];
var PutGroup = [
    9,
    n0,
    _PGu,
    {
        [_ht]: ["PUT", "/applications/{applicationId}/indices/{indexId}/groups", 200],
    },
    () => PutGroupRequest,
    () => PutGroupResponse,
];
var SearchRelevantContent = [
    9,
    n0,
    _SRC,
    {
        [_ht]: ["POST", "/applications/{applicationId}/relevant-content", 200],
    },
    () => SearchRelevantContentRequest,
    () => SearchRelevantContentResponse,
];
var StartDataSourceSyncJob = [
    9,
    n0,
    _SDSSJ,
    {
        [_ht]: ["POST", "/applications/{applicationId}/indices/{indexId}/datasources/{dataSourceId}/startsync", 200],
    },
    () => StartDataSourceSyncJobRequest,
    () => StartDataSourceSyncJobResponse,
];
var StopDataSourceSyncJob = [
    9,
    n0,
    _SDSSJt,
    {
        [_ht]: ["POST", "/applications/{applicationId}/indices/{indexId}/datasources/{dataSourceId}/stopsync", 200],
    },
    () => StopDataSourceSyncJobRequest,
    () => StopDataSourceSyncJobResponse,
];
var TagResource = [
    9,
    n0,
    _TR,
    {
        [_ht]: ["POST", "/v1/tags/{resourceARN}", 200],
    },
    () => TagResourceRequest,
    () => TagResourceResponse,
];
var UntagResource = [
    9,
    n0,
    _UR,
    {
        [_ht]: ["DELETE", "/v1/tags/{resourceARN}", 200],
    },
    () => UntagResourceRequest,
    () => UntagResourceResponse,
];
var UpdateApplication = [
    9,
    n0,
    _UAp,
    {
        [_ht]: ["PUT", "/applications/{applicationId}", 200],
    },
    () => UpdateApplicationRequest,
    () => UpdateApplicationResponse,
];
var UpdateChatControlsConfiguration = [
    9,
    n0,
    _UCCC,
    {
        [_ht]: ["PATCH", "/applications/{applicationId}/chatcontrols", 200],
    },
    () => UpdateChatControlsConfigurationRequest,
    () => UpdateChatControlsConfigurationResponse,
];
var UpdateChatResponseConfiguration = [
    9,
    n0,
    _UCRC,
    {
        [_ht]: ["PUT", "/applications/{applicationId}/chatresponseconfigurations/{chatResponseConfigurationId}", 200],
    },
    () => UpdateChatResponseConfigurationRequest,
    () => UpdateChatResponseConfigurationResponse,
];
var UpdateDataAccessor = [
    9,
    n0,
    _UDA,
    {
        [_ht]: ["PUT", "/applications/{applicationId}/dataaccessors/{dataAccessorId}", 200],
    },
    () => UpdateDataAccessorRequest,
    () => UpdateDataAccessorResponse,
];
var UpdateDataSource = [
    9,
    n0,
    _UDS,
    {
        [_ht]: ["PUT", "/applications/{applicationId}/indices/{indexId}/datasources/{dataSourceId}", 200],
    },
    () => UpdateDataSourceRequest,
    () => UpdateDataSourceResponse,
];
var UpdateIndex = [
    9,
    n0,
    _UI,
    {
        [_ht]: ["PUT", "/applications/{applicationId}/indices/{indexId}", 200],
    },
    () => UpdateIndexRequest,
    () => UpdateIndexResponse,
];
var UpdatePlugin = [
    9,
    n0,
    _UP,
    {
        [_ht]: ["PUT", "/applications/{applicationId}/plugins/{pluginId}", 200],
    },
    () => UpdatePluginRequest,
    () => UpdatePluginResponse,
];
var UpdateRetriever = [
    9,
    n0,
    _URp,
    {
        [_ht]: ["PUT", "/applications/{applicationId}/retrievers/{retrieverId}", 200],
    },
    () => UpdateRetrieverRequest,
    () => UpdateRetrieverResponse,
];
var UpdateSubscription = [
    9,
    n0,
    _US,
    {
        [_ht]: ["PUT", "/applications/{applicationId}/subscriptions/{subscriptionId}", 200],
    },
    () => UpdateSubscriptionRequest,
    () => UpdateSubscriptionResponse,
];
var UpdateUser = [
    9,
    n0,
    _UU,
    {
        [_ht]: ["PUT", "/applications/{applicationId}/users/{userId}", 200],
    },
    () => UpdateUserRequest,
    () => UpdateUserResponse,
];
var UpdateWebExperience = [
    9,
    n0,
    _UWE,
    {
        [_ht]: ["PUT", "/applications/{applicationId}/experiences/{webExperienceId}", 200],
    },
    () => UpdateWebExperienceRequest,
    () => UpdateWebExperienceResponse,
];

class AssociatePermissionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "AssociatePermission", {})
    .n("QBusinessClient", "AssociatePermissionCommand")
    .sc(AssociatePermission)
    .build() {
}

class BatchDeleteDocumentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "BatchDeleteDocument", {})
    .n("QBusinessClient", "BatchDeleteDocumentCommand")
    .sc(BatchDeleteDocument)
    .build() {
}

class BatchPutDocumentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "BatchPutDocument", {})
    .n("QBusinessClient", "BatchPutDocumentCommand")
    .sc(BatchPutDocument)
    .build() {
}

class CancelSubscriptionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "CancelSubscription", {})
    .n("QBusinessClient", "CancelSubscriptionCommand")
    .sc(CancelSubscription)
    .build() {
}

class ChatCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions()), middlewareEventstream.getEventStreamPlugin(config)];
})
    .s("ExpertQ", "Chat", {
    eventStream: {
        input: true,
        output: true,
    },
})
    .n("QBusinessClient", "ChatCommand")
    .sc(Chat)
    .build() {
}

class ChatSyncCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "ChatSync", {})
    .n("QBusinessClient", "ChatSyncCommand")
    .sc(ChatSync)
    .build() {
}

class CheckDocumentAccessCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "CheckDocumentAccess", {})
    .n("QBusinessClient", "CheckDocumentAccessCommand")
    .sc(CheckDocumentAccess)
    .build() {
}

class CreateAnonymousWebExperienceUrlCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "CreateAnonymousWebExperienceUrl", {})
    .n("QBusinessClient", "CreateAnonymousWebExperienceUrlCommand")
    .sc(CreateAnonymousWebExperienceUrl)
    .build() {
}

class CreateApplicationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "CreateApplication", {})
    .n("QBusinessClient", "CreateApplicationCommand")
    .sc(CreateApplication)
    .build() {
}

class CreateChatResponseConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "CreateChatResponseConfiguration", {})
    .n("QBusinessClient", "CreateChatResponseConfigurationCommand")
    .sc(CreateChatResponseConfiguration)
    .build() {
}

class CreateDataAccessorCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "CreateDataAccessor", {})
    .n("QBusinessClient", "CreateDataAccessorCommand")
    .sc(CreateDataAccessor)
    .build() {
}

class CreateDataSourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "CreateDataSource", {})
    .n("QBusinessClient", "CreateDataSourceCommand")
    .sc(CreateDataSource)
    .build() {
}

class CreateIndexCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "CreateIndex", {})
    .n("QBusinessClient", "CreateIndexCommand")
    .sc(CreateIndex)
    .build() {
}

class CreatePluginCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "CreatePlugin", {})
    .n("QBusinessClient", "CreatePluginCommand")
    .sc(CreatePlugin)
    .build() {
}

class CreateRetrieverCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "CreateRetriever", {})
    .n("QBusinessClient", "CreateRetrieverCommand")
    .sc(CreateRetriever)
    .build() {
}

class CreateSubscriptionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "CreateSubscription", {})
    .n("QBusinessClient", "CreateSubscriptionCommand")
    .sc(CreateSubscription)
    .build() {
}

class CreateUserCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "CreateUser", {})
    .n("QBusinessClient", "CreateUserCommand")
    .sc(CreateUser)
    .build() {
}

class CreateWebExperienceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "CreateWebExperience", {})
    .n("QBusinessClient", "CreateWebExperienceCommand")
    .sc(CreateWebExperience)
    .build() {
}

class DeleteApplicationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "DeleteApplication", {})
    .n("QBusinessClient", "DeleteApplicationCommand")
    .sc(DeleteApplication)
    .build() {
}

class DeleteAttachmentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "DeleteAttachment", {})
    .n("QBusinessClient", "DeleteAttachmentCommand")
    .sc(DeleteAttachment)
    .build() {
}

class DeleteChatControlsConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "DeleteChatControlsConfiguration", {})
    .n("QBusinessClient", "DeleteChatControlsConfigurationCommand")
    .sc(DeleteChatControlsConfiguration)
    .build() {
}

class DeleteChatResponseConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "DeleteChatResponseConfiguration", {})
    .n("QBusinessClient", "DeleteChatResponseConfigurationCommand")
    .sc(DeleteChatResponseConfiguration)
    .build() {
}

class DeleteConversationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "DeleteConversation", {})
    .n("QBusinessClient", "DeleteConversationCommand")
    .sc(DeleteConversation)
    .build() {
}

class DeleteDataAccessorCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "DeleteDataAccessor", {})
    .n("QBusinessClient", "DeleteDataAccessorCommand")
    .sc(DeleteDataAccessor)
    .build() {
}

class DeleteDataSourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "DeleteDataSource", {})
    .n("QBusinessClient", "DeleteDataSourceCommand")
    .sc(DeleteDataSource)
    .build() {
}

class DeleteGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "DeleteGroup", {})
    .n("QBusinessClient", "DeleteGroupCommand")
    .sc(DeleteGroup)
    .build() {
}

class DeleteIndexCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "DeleteIndex", {})
    .n("QBusinessClient", "DeleteIndexCommand")
    .sc(DeleteIndex)
    .build() {
}

class DeletePluginCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "DeletePlugin", {})
    .n("QBusinessClient", "DeletePluginCommand")
    .sc(DeletePlugin)
    .build() {
}

class DeleteRetrieverCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "DeleteRetriever", {})
    .n("QBusinessClient", "DeleteRetrieverCommand")
    .sc(DeleteRetriever)
    .build() {
}

class DeleteUserCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "DeleteUser", {})
    .n("QBusinessClient", "DeleteUserCommand")
    .sc(DeleteUser)
    .build() {
}

class DeleteWebExperienceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "DeleteWebExperience", {})
    .n("QBusinessClient", "DeleteWebExperienceCommand")
    .sc(DeleteWebExperience)
    .build() {
}

class DisassociatePermissionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "DisassociatePermission", {})
    .n("QBusinessClient", "DisassociatePermissionCommand")
    .sc(DisassociatePermission)
    .build() {
}

class GetApplicationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "GetApplication", {})
    .n("QBusinessClient", "GetApplicationCommand")
    .sc(GetApplication)
    .build() {
}

class GetChatControlsConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "GetChatControlsConfiguration", {})
    .n("QBusinessClient", "GetChatControlsConfigurationCommand")
    .sc(GetChatControlsConfiguration)
    .build() {
}

class GetChatResponseConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "GetChatResponseConfiguration", {})
    .n("QBusinessClient", "GetChatResponseConfigurationCommand")
    .sc(GetChatResponseConfiguration)
    .build() {
}

class GetDataAccessorCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "GetDataAccessor", {})
    .n("QBusinessClient", "GetDataAccessorCommand")
    .sc(GetDataAccessor)
    .build() {
}

class GetDataSourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "GetDataSource", {})
    .n("QBusinessClient", "GetDataSourceCommand")
    .sc(GetDataSource)
    .build() {
}

class GetDocumentContentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "GetDocumentContent", {})
    .n("QBusinessClient", "GetDocumentContentCommand")
    .sc(GetDocumentContent)
    .build() {
}

class GetGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "GetGroup", {})
    .n("QBusinessClient", "GetGroupCommand")
    .sc(GetGroup)
    .build() {
}

class GetIndexCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "GetIndex", {})
    .n("QBusinessClient", "GetIndexCommand")
    .sc(GetIndex)
    .build() {
}

class GetMediaCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "GetMedia", {})
    .n("QBusinessClient", "GetMediaCommand")
    .sc(GetMedia)
    .build() {
}

class GetPluginCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "GetPlugin", {})
    .n("QBusinessClient", "GetPluginCommand")
    .sc(GetPlugin)
    .build() {
}

class GetPolicyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "GetPolicy", {})
    .n("QBusinessClient", "GetPolicyCommand")
    .sc(GetPolicy)
    .build() {
}

class GetRetrieverCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "GetRetriever", {})
    .n("QBusinessClient", "GetRetrieverCommand")
    .sc(GetRetriever)
    .build() {
}

class GetUserCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "GetUser", {})
    .n("QBusinessClient", "GetUserCommand")
    .sc(GetUser)
    .build() {
}

class GetWebExperienceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "GetWebExperience", {})
    .n("QBusinessClient", "GetWebExperienceCommand")
    .sc(GetWebExperience)
    .build() {
}

class ListApplicationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "ListApplications", {})
    .n("QBusinessClient", "ListApplicationsCommand")
    .sc(ListApplications)
    .build() {
}

class ListAttachmentsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "ListAttachments", {})
    .n("QBusinessClient", "ListAttachmentsCommand")
    .sc(ListAttachments)
    .build() {
}

class ListChatResponseConfigurationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "ListChatResponseConfigurations", {})
    .n("QBusinessClient", "ListChatResponseConfigurationsCommand")
    .sc(ListChatResponseConfigurations)
    .build() {
}

class ListConversationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "ListConversations", {})
    .n("QBusinessClient", "ListConversationsCommand")
    .sc(ListConversations)
    .build() {
}

class ListDataAccessorsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "ListDataAccessors", {})
    .n("QBusinessClient", "ListDataAccessorsCommand")
    .sc(ListDataAccessors)
    .build() {
}

class ListDataSourcesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "ListDataSources", {})
    .n("QBusinessClient", "ListDataSourcesCommand")
    .sc(ListDataSources)
    .build() {
}

class ListDataSourceSyncJobsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "ListDataSourceSyncJobs", {})
    .n("QBusinessClient", "ListDataSourceSyncJobsCommand")
    .sc(ListDataSourceSyncJobs)
    .build() {
}

class ListDocumentsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "ListDocuments", {})
    .n("QBusinessClient", "ListDocumentsCommand")
    .sc(ListDocuments)
    .build() {
}

class ListGroupsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "ListGroups", {})
    .n("QBusinessClient", "ListGroupsCommand")
    .sc(ListGroups)
    .build() {
}

class ListIndicesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "ListIndices", {})
    .n("QBusinessClient", "ListIndicesCommand")
    .sc(ListIndices)
    .build() {
}

class ListMessagesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "ListMessages", {})
    .n("QBusinessClient", "ListMessagesCommand")
    .sc(ListMessages)
    .build() {
}

class ListPluginActionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "ListPluginActions", {})
    .n("QBusinessClient", "ListPluginActionsCommand")
    .sc(ListPluginActions)
    .build() {
}

class ListPluginsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "ListPlugins", {})
    .n("QBusinessClient", "ListPluginsCommand")
    .sc(ListPlugins)
    .build() {
}

class ListPluginTypeActionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "ListPluginTypeActions", {})
    .n("QBusinessClient", "ListPluginTypeActionsCommand")
    .sc(ListPluginTypeActions)
    .build() {
}

class ListPluginTypeMetadataCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "ListPluginTypeMetadata", {})
    .n("QBusinessClient", "ListPluginTypeMetadataCommand")
    .sc(ListPluginTypeMetadata)
    .build() {
}

class ListRetrieversCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "ListRetrievers", {})
    .n("QBusinessClient", "ListRetrieversCommand")
    .sc(ListRetrievers)
    .build() {
}

class ListSubscriptionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "ListSubscriptions", {})
    .n("QBusinessClient", "ListSubscriptionsCommand")
    .sc(ListSubscriptions)
    .build() {
}

class ListTagsForResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "ListTagsForResource", {})
    .n("QBusinessClient", "ListTagsForResourceCommand")
    .sc(ListTagsForResource)
    .build() {
}

class ListWebExperiencesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "ListWebExperiences", {})
    .n("QBusinessClient", "ListWebExperiencesCommand")
    .sc(ListWebExperiences)
    .build() {
}

class PutFeedbackCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "PutFeedback", {})
    .n("QBusinessClient", "PutFeedbackCommand")
    .sc(PutFeedback)
    .build() {
}

class PutGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "PutGroup", {})
    .n("QBusinessClient", "PutGroupCommand")
    .sc(PutGroup)
    .build() {
}

class SearchRelevantContentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "SearchRelevantContent", {})
    .n("QBusinessClient", "SearchRelevantContentCommand")
    .sc(SearchRelevantContent)
    .build() {
}

class StartDataSourceSyncJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "StartDataSourceSyncJob", {})
    .n("QBusinessClient", "StartDataSourceSyncJobCommand")
    .sc(StartDataSourceSyncJob)
    .build() {
}

class StopDataSourceSyncJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "StopDataSourceSyncJob", {})
    .n("QBusinessClient", "StopDataSourceSyncJobCommand")
    .sc(StopDataSourceSyncJob)
    .build() {
}

class TagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "TagResource", {})
    .n("QBusinessClient", "TagResourceCommand")
    .sc(TagResource)
    .build() {
}

class UntagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "UntagResource", {})
    .n("QBusinessClient", "UntagResourceCommand")
    .sc(UntagResource)
    .build() {
}

class UpdateApplicationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "UpdateApplication", {})
    .n("QBusinessClient", "UpdateApplicationCommand")
    .sc(UpdateApplication)
    .build() {
}

class UpdateChatControlsConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "UpdateChatControlsConfiguration", {})
    .n("QBusinessClient", "UpdateChatControlsConfigurationCommand")
    .sc(UpdateChatControlsConfiguration)
    .build() {
}

class UpdateChatResponseConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "UpdateChatResponseConfiguration", {})
    .n("QBusinessClient", "UpdateChatResponseConfigurationCommand")
    .sc(UpdateChatResponseConfiguration)
    .build() {
}

class UpdateDataAccessorCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "UpdateDataAccessor", {})
    .n("QBusinessClient", "UpdateDataAccessorCommand")
    .sc(UpdateDataAccessor)
    .build() {
}

class UpdateDataSourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "UpdateDataSource", {})
    .n("QBusinessClient", "UpdateDataSourceCommand")
    .sc(UpdateDataSource)
    .build() {
}

class UpdateIndexCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "UpdateIndex", {})
    .n("QBusinessClient", "UpdateIndexCommand")
    .sc(UpdateIndex)
    .build() {
}

class UpdatePluginCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "UpdatePlugin", {})
    .n("QBusinessClient", "UpdatePluginCommand")
    .sc(UpdatePlugin)
    .build() {
}

class UpdateRetrieverCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "UpdateRetriever", {})
    .n("QBusinessClient", "UpdateRetrieverCommand")
    .sc(UpdateRetriever)
    .build() {
}

class UpdateSubscriptionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "UpdateSubscription", {})
    .n("QBusinessClient", "UpdateSubscriptionCommand")
    .sc(UpdateSubscription)
    .build() {
}

class UpdateUserCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "UpdateUser", {})
    .n("QBusinessClient", "UpdateUserCommand")
    .sc(UpdateUser)
    .build() {
}

class UpdateWebExperienceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("ExpertQ", "UpdateWebExperience", {})
    .n("QBusinessClient", "UpdateWebExperienceCommand")
    .sc(UpdateWebExperience)
    .build() {
}

const commands = {
    AssociatePermissionCommand,
    BatchDeleteDocumentCommand,
    BatchPutDocumentCommand,
    CancelSubscriptionCommand,
    ChatCommand,
    ChatSyncCommand,
    CheckDocumentAccessCommand,
    CreateAnonymousWebExperienceUrlCommand,
    CreateApplicationCommand,
    CreateChatResponseConfigurationCommand,
    CreateDataAccessorCommand,
    CreateDataSourceCommand,
    CreateIndexCommand,
    CreatePluginCommand,
    CreateRetrieverCommand,
    CreateSubscriptionCommand,
    CreateUserCommand,
    CreateWebExperienceCommand,
    DeleteApplicationCommand,
    DeleteAttachmentCommand,
    DeleteChatControlsConfigurationCommand,
    DeleteChatResponseConfigurationCommand,
    DeleteConversationCommand,
    DeleteDataAccessorCommand,
    DeleteDataSourceCommand,
    DeleteGroupCommand,
    DeleteIndexCommand,
    DeletePluginCommand,
    DeleteRetrieverCommand,
    DeleteUserCommand,
    DeleteWebExperienceCommand,
    DisassociatePermissionCommand,
    GetApplicationCommand,
    GetChatControlsConfigurationCommand,
    GetChatResponseConfigurationCommand,
    GetDataAccessorCommand,
    GetDataSourceCommand,
    GetDocumentContentCommand,
    GetGroupCommand,
    GetIndexCommand,
    GetMediaCommand,
    GetPluginCommand,
    GetPolicyCommand,
    GetRetrieverCommand,
    GetUserCommand,
    GetWebExperienceCommand,
    ListApplicationsCommand,
    ListAttachmentsCommand,
    ListChatResponseConfigurationsCommand,
    ListConversationsCommand,
    ListDataAccessorsCommand,
    ListDataSourcesCommand,
    ListDataSourceSyncJobsCommand,
    ListDocumentsCommand,
    ListGroupsCommand,
    ListIndicesCommand,
    ListMessagesCommand,
    ListPluginActionsCommand,
    ListPluginsCommand,
    ListPluginTypeActionsCommand,
    ListPluginTypeMetadataCommand,
    ListRetrieversCommand,
    ListSubscriptionsCommand,
    ListTagsForResourceCommand,
    ListWebExperiencesCommand,
    PutFeedbackCommand,
    PutGroupCommand,
    SearchRelevantContentCommand,
    StartDataSourceSyncJobCommand,
    StopDataSourceSyncJobCommand,
    TagResourceCommand,
    UntagResourceCommand,
    UpdateApplicationCommand,
    UpdateChatControlsConfigurationCommand,
    UpdateChatResponseConfigurationCommand,
    UpdateDataAccessorCommand,
    UpdateDataSourceCommand,
    UpdateIndexCommand,
    UpdatePluginCommand,
    UpdateRetrieverCommand,
    UpdateSubscriptionCommand,
    UpdateUserCommand,
    UpdateWebExperienceCommand,
};
class QBusiness extends QBusinessClient {
}
smithyClient.createAggregatedClient(commands, QBusiness);

const paginateGetChatControlsConfiguration = core.createPaginator(QBusinessClient, GetChatControlsConfigurationCommand, "nextToken", "nextToken", "maxResults");

const paginateListApplications = core.createPaginator(QBusinessClient, ListApplicationsCommand, "nextToken", "nextToken", "maxResults");

const paginateListAttachments = core.createPaginator(QBusinessClient, ListAttachmentsCommand, "nextToken", "nextToken", "maxResults");

const paginateListChatResponseConfigurations = core.createPaginator(QBusinessClient, ListChatResponseConfigurationsCommand, "nextToken", "nextToken", "maxResults");

const paginateListConversations = core.createPaginator(QBusinessClient, ListConversationsCommand, "nextToken", "nextToken", "maxResults");

const paginateListDataAccessors = core.createPaginator(QBusinessClient, ListDataAccessorsCommand, "nextToken", "nextToken", "maxResults");

const paginateListDataSourceSyncJobs = core.createPaginator(QBusinessClient, ListDataSourceSyncJobsCommand, "nextToken", "nextToken", "maxResults");

const paginateListDataSources = core.createPaginator(QBusinessClient, ListDataSourcesCommand, "nextToken", "nextToken", "maxResults");

const paginateListDocuments = core.createPaginator(QBusinessClient, ListDocumentsCommand, "nextToken", "nextToken", "maxResults");

const paginateListGroups = core.createPaginator(QBusinessClient, ListGroupsCommand, "nextToken", "nextToken", "maxResults");

const paginateListIndices = core.createPaginator(QBusinessClient, ListIndicesCommand, "nextToken", "nextToken", "maxResults");

const paginateListMessages = core.createPaginator(QBusinessClient, ListMessagesCommand, "nextToken", "nextToken", "maxResults");

const paginateListPluginActions = core.createPaginator(QBusinessClient, ListPluginActionsCommand, "nextToken", "nextToken", "maxResults");

const paginateListPluginTypeActions = core.createPaginator(QBusinessClient, ListPluginTypeActionsCommand, "nextToken", "nextToken", "maxResults");

const paginateListPluginTypeMetadata = core.createPaginator(QBusinessClient, ListPluginTypeMetadataCommand, "nextToken", "nextToken", "maxResults");

const paginateListPlugins = core.createPaginator(QBusinessClient, ListPluginsCommand, "nextToken", "nextToken", "maxResults");

const paginateListRetrievers = core.createPaginator(QBusinessClient, ListRetrieversCommand, "nextToken", "nextToken", "maxResults");

const paginateListSubscriptions = core.createPaginator(QBusinessClient, ListSubscriptionsCommand, "nextToken", "nextToken", "maxResults");

const paginateListWebExperiences = core.createPaginator(QBusinessClient, ListWebExperiencesCommand, "nextToken", "nextToken", "maxResults");

const paginateSearchRelevantContent = core.createPaginator(QBusinessClient, SearchRelevantContentCommand, "nextToken", "nextToken", "maxResults");

const MemberRelation = {
    AND: "AND",
    OR: "OR",
};
const ReadAccessType = {
    ALLOW: "ALLOW",
    DENY: "DENY",
};
const MembershipType = {
    DATASOURCE: "DATASOURCE",
    INDEX: "INDEX",
};
const ActionPayloadFieldType = {
    ARRAY: "ARRAY",
    BOOLEAN: "BOOLEAN",
    NUMBER: "NUMBER",
    STRING: "STRING",
};
const PluginType = {
    ASANA: "ASANA",
    ATLASSIAN_CONFLUENCE: "ATLASSIAN_CONFLUENCE",
    CUSTOM: "CUSTOM",
    GOOGLE_CALENDAR: "GOOGLE_CALENDAR",
    JIRA: "JIRA",
    JIRA_CLOUD: "JIRA_CLOUD",
    MICROSOFT_EXCHANGE: "MICROSOFT_EXCHANGE",
    MICROSOFT_TEAMS: "MICROSOFT_TEAMS",
    PAGERDUTY_ADVANCE: "PAGERDUTY_ADVANCE",
    QUICKSIGHT: "QUICKSIGHT",
    SALESFORCE: "SALESFORCE",
    SALESFORCE_CRM: "SALESFORCE_CRM",
    SERVICENOW_NOW_PLATFORM: "SERVICENOW_NOW_PLATFORM",
    SERVICE_NOW: "SERVICE_NOW",
    SMARTSHEET: "SMARTSHEET",
    ZENDESK: "ZENDESK",
    ZENDESK_SUITE: "ZENDESK_SUITE",
};
const APISchemaType = {
    OPEN_API_V3: "OPEN_API_V3",
};
const IdentityType = {
    ANONYMOUS: "ANONYMOUS",
    AWS_IAM_IDC: "AWS_IAM_IDC",
    AWS_IAM_IDP_OIDC: "AWS_IAM_IDP_OIDC",
    AWS_IAM_IDP_SAML: "AWS_IAM_IDP_SAML",
    AWS_QUICKSIGHT_IDP: "AWS_QUICKSIGHT_IDP",
};
const ApplicationStatus = {
    ACTIVE: "ACTIVE",
    CREATING: "CREATING",
    DELETING: "DELETING",
    FAILED: "FAILED",
    UPDATING: "UPDATING",
};
const AttachmentsControlMode = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const PersonalizationControlMode = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const QAppsControlMode = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const ValidationExceptionReason = {
    CANNOT_PARSE: "CANNOT_PARSE",
    FIELD_VALIDATION_FAILED: "FIELD_VALIDATION_FAILED",
    UNKNOWN_OPERATION: "UNKNOWN_OPERATION",
};
const DataAccessorAuthenticationType = {
    AWS_IAM_IDC_AUTH_CODE: "AWS_IAM_IDC_AUTH_CODE",
    AWS_IAM_IDC_TTI: "AWS_IAM_IDC_TTI",
};
const AutoSubscriptionStatus = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const SubscriptionType = {
    Q_BUSINESS: "Q_BUSINESS",
    Q_LITE: "Q_LITE",
};
const ErrorCode = {
    INTERNAL_ERROR: "InternalError",
    INVALID_REQUEST: "InvalidRequest",
    RESOURCE_INACTIVE: "ResourceInactive",
    RESOURCE_NOT_FOUND: "ResourceNotFound",
};
const IndexType = {
    ENTERPRISE: "ENTERPRISE",
    STARTER: "STARTER",
};
const DocumentEnrichmentConditionOperator = {
    BEGINS_WITH: "BEGINS_WITH",
    CONTAINS: "CONTAINS",
    EQUALS: "EQUALS",
    EXISTS: "EXISTS",
    GREATER_THAN: "GREATER_THAN",
    GREATER_THAN_OR_EQUALS: "GREATER_THAN_OR_EQUALS",
    LESS_THAN: "LESS_THAN",
    LESS_THAN_OR_EQUALS: "LESS_THAN_OR_EQUALS",
    NOT_CONTAINS: "NOT_CONTAINS",
    NOT_EQUALS: "NOT_EQUALS",
    NOT_EXISTS: "NOT_EXISTS",
};
const DocumentContentOperator = {
    DELETE: "DELETE",
};
const AttributeValueOperator = {
    DELETE: "DELETE",
};
const AudioExtractionStatus = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const ImageExtractionStatus = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const VideoExtractionStatus = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const DataSourceStatus = {
    ACTIVE: "ACTIVE",
    CREATING: "CREATING",
    DELETING: "DELETING",
    FAILED: "FAILED",
    PENDING_CREATION: "PENDING_CREATION",
    UPDATING: "UPDATING",
};
const Status = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const AttributeType = {
    DATE: "DATE",
    NUMBER: "NUMBER",
    STRING: "STRING",
    STRING_LIST: "STRING_LIST",
};
const IndexStatus = {
    ACTIVE: "ACTIVE",
    CREATING: "CREATING",
    DELETING: "DELETING",
    FAILED: "FAILED",
    UPDATING: "UPDATING",
};
const PluginBuildStatus = {
    CREATE_FAILED: "CREATE_FAILED",
    CREATE_IN_PROGRESS: "CREATE_IN_PROGRESS",
    DELETE_FAILED: "DELETE_FAILED",
    DELETE_IN_PROGRESS: "DELETE_IN_PROGRESS",
    READY: "READY",
    UPDATE_FAILED: "UPDATE_FAILED",
    UPDATE_IN_PROGRESS: "UPDATE_IN_PROGRESS",
};
const PluginState = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const DocumentAttributeBoostingLevel = {
    HIGH: "HIGH",
    LOW: "LOW",
    MEDIUM: "MEDIUM",
    NONE: "NONE",
    ONE: "ONE",
    TWO: "TWO",
    VERY_HIGH: "VERY_HIGH",
};
const NumberAttributeBoostingType = {
    PRIORITIZE_LARGER_VALUES: "PRIORITIZE_LARGER_VALUES",
    PRIORITIZE_SMALLER_VALUES: "PRIORITIZE_SMALLER_VALUES",
};
const StringAttributeValueBoostingLevel = {
    FIVE: "FIVE",
    FOUR: "FOUR",
    HIGH: "HIGH",
    LOW: "LOW",
    MEDIUM: "MEDIUM",
    ONE: "ONE",
    THREE: "THREE",
    TWO: "TWO",
    VERY_HIGH: "VERY_HIGH",
};
const RetrieverType = {
    KENDRA_INDEX: "KENDRA_INDEX",
    NATIVE_INDEX: "NATIVE_INDEX",
};
const RetrieverStatus = {
    ACTIVE: "ACTIVE",
    CREATING: "CREATING",
    FAILED: "FAILED",
};
const BrowserExtension = {
    CHROME: "CHROME",
    FIREFOX: "FIREFOX",
};
const WebExperienceSamplePromptsControlMode = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const WebExperienceStatus = {
    ACTIVE: "ACTIVE",
    CREATING: "CREATING",
    DELETING: "DELETING",
    FAILED: "FAILED",
    PENDING_AUTH_CONFIG: "PENDING_AUTH_CONFIG",
};
const CreatorModeControl = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const OrchestrationControl = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const PermissionConditionOperator = {
    STRING_EQUALS: "StringEquals",
};
const AttachmentStatus = {
    FAILED: "FAILED",
    SUCCESS: "SUCCESS",
};
const AudioExtractionType = {
    SUMMARY: "SUMMARY",
    TRANSCRIPT: "TRANSCRIPT",
};
const ContentType = {
    CSV: "CSV",
    HTML: "HTML",
    JSON: "JSON",
    MD: "MD",
    MS_EXCEL: "MS_EXCEL",
    MS_WORD: "MS_WORD",
    PDF: "PDF",
    PLAIN_TEXT: "PLAIN_TEXT",
    PPT: "PPT",
    RTF: "RTF",
    XML: "XML",
    XSLT: "XSLT",
};
const ChatMode = {
    CREATOR_MODE: "CREATOR_MODE",
    PLUGIN_MODE: "PLUGIN_MODE",
    RETRIEVAL_MODE: "RETRIEVAL_MODE",
};
const VideoExtractionType = {
    SUMMARY: "SUMMARY",
    TRANSCRIPT: "TRANSCRIPT",
};
const SystemMessageType = {
    GROUNDED_RESPONSE: "GROUNDED_RESPONSE",
    RESPONSE: "RESPONSE",
};
const ChatResponseConfigurationStatus = {
    ACTIVE: "ACTIVE",
    CREATING: "CREATING",
    FAILED: "FAILED",
    UPDATING: "UPDATING",
};
const ResponseConfigurationType = {
    ALL: "ALL",
};
const DataSourceSyncJobStatus = {
    ABORTED: "ABORTED",
    FAILED: "FAILED",
    INCOMPLETE: "INCOMPLETE",
    STOPPING: "STOPPING",
    SUCCEEDED: "SUCCEEDED",
    SYNCING: "SYNCING",
    SYNCING_INDEXING: "SYNCING_INDEXING",
};
const DocumentStatus = {
    DELETED: "DELETED",
    DELETING: "DELETING",
    DOCUMENT_FAILED_TO_INDEX: "DOCUMENT_FAILED_TO_INDEX",
    FAILED: "FAILED",
    INDEXED: "INDEXED",
    PROCESSING: "PROCESSING",
    RECEIVED: "RECEIVED",
    UPDATED: "UPDATED",
};
const HallucinationReductionControl = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const ResponseScope = {
    ENTERPRISE_CONTENT_ONLY: "ENTERPRISE_CONTENT_ONLY",
    EXTENDED_KNOWLEDGE_ENABLED: "EXTENDED_KNOWLEDGE_ENABLED",
};
const RuleType = {
    CONTENT_BLOCKER_RULE: "CONTENT_BLOCKER_RULE",
    CONTENT_RETRIEVAL_RULE: "CONTENT_RETRIEVAL_RULE",
};
const OutputFormat = {
    EXTRACTED: "EXTRACTED",
    RAW: "RAW",
};
const GroupStatus = {
    DELETED: "DELETED",
    DELETING: "DELETING",
    FAILED: "FAILED",
    PROCESSING: "PROCESSING",
    SUCCEEDED: "SUCCEEDED",
};
const MessageType = {
    SYSTEM: "SYSTEM",
    USER: "USER",
};
const PluginTypeCategory = {
    COMMUNICATION: "Communication",
    CRM: "Customer relationship management (CRM)",
    PRODUCTIVITY: "Productivity",
    PROJECT_MANAGEMENT: "Project management",
    TICKETING_MANAGEMENT: "Ticketing and incident management",
};
const MessageUsefulnessReason = {
    COMPLETE: "COMPLETE",
    FACTUALLY_CORRECT: "FACTUALLY_CORRECT",
    HARMFUL_OR_UNSAFE: "HARMFUL_OR_UNSAFE",
    HELPFUL: "HELPFUL",
    INCORRECT_OR_MISSING_SOURCES: "INCORRECT_OR_MISSING_SOURCES",
    NOT_BASED_ON_DOCUMENTS: "NOT_BASED_ON_DOCUMENTS",
    NOT_COMPLETE: "NOT_COMPLETE",
    NOT_CONCISE: "NOT_CONCISE",
    NOT_FACTUALLY_CORRECT: "NOT_FACTUALLY_CORRECT",
    NOT_HELPFUL: "NOT_HELPFUL",
    OTHER: "OTHER",
    RELEVANT_SOURCES: "RELEVANT_SOURCES",
};
const MessageUsefulness = {
    NOT_USEFUL: "NOT_USEFUL",
    USEFUL: "USEFUL",
};
const ScoreConfidence = {
    HIGH: "HIGH",
    LOW: "LOW",
    MEDIUM: "MEDIUM",
    NOT_AVAILABLE: "NOT_AVAILABLE",
    VERY_HIGH: "VERY_HIGH",
};

Object.defineProperty(exports, "$Command", {
    enumerable: true,
    get: function () { return smithyClient.Command; }
});
Object.defineProperty(exports, "__Client", {
    enumerable: true,
    get: function () { return smithyClient.Client; }
});
exports.APISchemaType = APISchemaType;
exports.AccessDeniedException = AccessDeniedException$1;
exports.ActionPayloadFieldType = ActionPayloadFieldType;
exports.ApplicationStatus = ApplicationStatus;
exports.AssociatePermissionCommand = AssociatePermissionCommand;
exports.AttachmentStatus = AttachmentStatus;
exports.AttachmentsControlMode = AttachmentsControlMode;
exports.AttributeType = AttributeType;
exports.AttributeValueOperator = AttributeValueOperator;
exports.AudioExtractionStatus = AudioExtractionStatus;
exports.AudioExtractionType = AudioExtractionType;
exports.AutoSubscriptionStatus = AutoSubscriptionStatus;
exports.BatchDeleteDocumentCommand = BatchDeleteDocumentCommand;
exports.BatchPutDocumentCommand = BatchPutDocumentCommand;
exports.BrowserExtension = BrowserExtension;
exports.CancelSubscriptionCommand = CancelSubscriptionCommand;
exports.ChatCommand = ChatCommand;
exports.ChatMode = ChatMode;
exports.ChatResponseConfigurationStatus = ChatResponseConfigurationStatus;
exports.ChatSyncCommand = ChatSyncCommand;
exports.CheckDocumentAccessCommand = CheckDocumentAccessCommand;
exports.ConflictException = ConflictException$1;
exports.ContentType = ContentType;
exports.CreateAnonymousWebExperienceUrlCommand = CreateAnonymousWebExperienceUrlCommand;
exports.CreateApplicationCommand = CreateApplicationCommand;
exports.CreateChatResponseConfigurationCommand = CreateChatResponseConfigurationCommand;
exports.CreateDataAccessorCommand = CreateDataAccessorCommand;
exports.CreateDataSourceCommand = CreateDataSourceCommand;
exports.CreateIndexCommand = CreateIndexCommand;
exports.CreatePluginCommand = CreatePluginCommand;
exports.CreateRetrieverCommand = CreateRetrieverCommand;
exports.CreateSubscriptionCommand = CreateSubscriptionCommand;
exports.CreateUserCommand = CreateUserCommand;
exports.CreateWebExperienceCommand = CreateWebExperienceCommand;
exports.CreatorModeControl = CreatorModeControl;
exports.DataAccessorAuthenticationType = DataAccessorAuthenticationType;
exports.DataSourceStatus = DataSourceStatus;
exports.DataSourceSyncJobStatus = DataSourceSyncJobStatus;
exports.DeleteApplicationCommand = DeleteApplicationCommand;
exports.DeleteAttachmentCommand = DeleteAttachmentCommand;
exports.DeleteChatControlsConfigurationCommand = DeleteChatControlsConfigurationCommand;
exports.DeleteChatResponseConfigurationCommand = DeleteChatResponseConfigurationCommand;
exports.DeleteConversationCommand = DeleteConversationCommand;
exports.DeleteDataAccessorCommand = DeleteDataAccessorCommand;
exports.DeleteDataSourceCommand = DeleteDataSourceCommand;
exports.DeleteGroupCommand = DeleteGroupCommand;
exports.DeleteIndexCommand = DeleteIndexCommand;
exports.DeletePluginCommand = DeletePluginCommand;
exports.DeleteRetrieverCommand = DeleteRetrieverCommand;
exports.DeleteUserCommand = DeleteUserCommand;
exports.DeleteWebExperienceCommand = DeleteWebExperienceCommand;
exports.DisassociatePermissionCommand = DisassociatePermissionCommand;
exports.DocumentAttributeBoostingLevel = DocumentAttributeBoostingLevel;
exports.DocumentContentOperator = DocumentContentOperator;
exports.DocumentEnrichmentConditionOperator = DocumentEnrichmentConditionOperator;
exports.DocumentStatus = DocumentStatus;
exports.ErrorCode = ErrorCode;
exports.ExternalResourceException = ExternalResourceException$1;
exports.GetApplicationCommand = GetApplicationCommand;
exports.GetChatControlsConfigurationCommand = GetChatControlsConfigurationCommand;
exports.GetChatResponseConfigurationCommand = GetChatResponseConfigurationCommand;
exports.GetDataAccessorCommand = GetDataAccessorCommand;
exports.GetDataSourceCommand = GetDataSourceCommand;
exports.GetDocumentContentCommand = GetDocumentContentCommand;
exports.GetGroupCommand = GetGroupCommand;
exports.GetIndexCommand = GetIndexCommand;
exports.GetMediaCommand = GetMediaCommand;
exports.GetPluginCommand = GetPluginCommand;
exports.GetPolicyCommand = GetPolicyCommand;
exports.GetRetrieverCommand = GetRetrieverCommand;
exports.GetUserCommand = GetUserCommand;
exports.GetWebExperienceCommand = GetWebExperienceCommand;
exports.GroupStatus = GroupStatus;
exports.HallucinationReductionControl = HallucinationReductionControl;
exports.IdentityType = IdentityType;
exports.ImageExtractionStatus = ImageExtractionStatus;
exports.IndexStatus = IndexStatus;
exports.IndexType = IndexType;
exports.InternalServerException = InternalServerException$1;
exports.LicenseNotFoundException = LicenseNotFoundException$1;
exports.ListApplicationsCommand = ListApplicationsCommand;
exports.ListAttachmentsCommand = ListAttachmentsCommand;
exports.ListChatResponseConfigurationsCommand = ListChatResponseConfigurationsCommand;
exports.ListConversationsCommand = ListConversationsCommand;
exports.ListDataAccessorsCommand = ListDataAccessorsCommand;
exports.ListDataSourceSyncJobsCommand = ListDataSourceSyncJobsCommand;
exports.ListDataSourcesCommand = ListDataSourcesCommand;
exports.ListDocumentsCommand = ListDocumentsCommand;
exports.ListGroupsCommand = ListGroupsCommand;
exports.ListIndicesCommand = ListIndicesCommand;
exports.ListMessagesCommand = ListMessagesCommand;
exports.ListPluginActionsCommand = ListPluginActionsCommand;
exports.ListPluginTypeActionsCommand = ListPluginTypeActionsCommand;
exports.ListPluginTypeMetadataCommand = ListPluginTypeMetadataCommand;
exports.ListPluginsCommand = ListPluginsCommand;
exports.ListRetrieversCommand = ListRetrieversCommand;
exports.ListSubscriptionsCommand = ListSubscriptionsCommand;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListWebExperiencesCommand = ListWebExperiencesCommand;
exports.MediaTooLargeException = MediaTooLargeException$1;
exports.MemberRelation = MemberRelation;
exports.MembershipType = MembershipType;
exports.MessageType = MessageType;
exports.MessageUsefulness = MessageUsefulness;
exports.MessageUsefulnessReason = MessageUsefulnessReason;
exports.NumberAttributeBoostingType = NumberAttributeBoostingType;
exports.OrchestrationControl = OrchestrationControl;
exports.OutputFormat = OutputFormat;
exports.PermissionConditionOperator = PermissionConditionOperator;
exports.PersonalizationControlMode = PersonalizationControlMode;
exports.PluginBuildStatus = PluginBuildStatus;
exports.PluginState = PluginState;
exports.PluginType = PluginType;
exports.PluginTypeCategory = PluginTypeCategory;
exports.PutFeedbackCommand = PutFeedbackCommand;
exports.PutGroupCommand = PutGroupCommand;
exports.QAppsControlMode = QAppsControlMode;
exports.QBusiness = QBusiness;
exports.QBusinessClient = QBusinessClient;
exports.QBusinessServiceException = QBusinessServiceException$1;
exports.ReadAccessType = ReadAccessType;
exports.ResourceNotFoundException = ResourceNotFoundException$1;
exports.ResponseConfigurationType = ResponseConfigurationType;
exports.ResponseScope = ResponseScope;
exports.RetrieverStatus = RetrieverStatus;
exports.RetrieverType = RetrieverType;
exports.RuleType = RuleType;
exports.ScoreConfidence = ScoreConfidence;
exports.SearchRelevantContentCommand = SearchRelevantContentCommand;
exports.ServiceQuotaExceededException = ServiceQuotaExceededException$1;
exports.StartDataSourceSyncJobCommand = StartDataSourceSyncJobCommand;
exports.Status = Status;
exports.StopDataSourceSyncJobCommand = StopDataSourceSyncJobCommand;
exports.StringAttributeValueBoostingLevel = StringAttributeValueBoostingLevel;
exports.SubscriptionType = SubscriptionType;
exports.SystemMessageType = SystemMessageType;
exports.TagResourceCommand = TagResourceCommand;
exports.ThrottlingException = ThrottlingException$1;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UpdateApplicationCommand = UpdateApplicationCommand;
exports.UpdateChatControlsConfigurationCommand = UpdateChatControlsConfigurationCommand;
exports.UpdateChatResponseConfigurationCommand = UpdateChatResponseConfigurationCommand;
exports.UpdateDataAccessorCommand = UpdateDataAccessorCommand;
exports.UpdateDataSourceCommand = UpdateDataSourceCommand;
exports.UpdateIndexCommand = UpdateIndexCommand;
exports.UpdatePluginCommand = UpdatePluginCommand;
exports.UpdateRetrieverCommand = UpdateRetrieverCommand;
exports.UpdateSubscriptionCommand = UpdateSubscriptionCommand;
exports.UpdateUserCommand = UpdateUserCommand;
exports.UpdateWebExperienceCommand = UpdateWebExperienceCommand;
exports.ValidationException = ValidationException$1;
exports.ValidationExceptionReason = ValidationExceptionReason;
exports.VideoExtractionStatus = VideoExtractionStatus;
exports.VideoExtractionType = VideoExtractionType;
exports.WebExperienceSamplePromptsControlMode = WebExperienceSamplePromptsControlMode;
exports.WebExperienceStatus = WebExperienceStatus;
exports.paginateGetChatControlsConfiguration = paginateGetChatControlsConfiguration;
exports.paginateListApplications = paginateListApplications;
exports.paginateListAttachments = paginateListAttachments;
exports.paginateListChatResponseConfigurations = paginateListChatResponseConfigurations;
exports.paginateListConversations = paginateListConversations;
exports.paginateListDataAccessors = paginateListDataAccessors;
exports.paginateListDataSourceSyncJobs = paginateListDataSourceSyncJobs;
exports.paginateListDataSources = paginateListDataSources;
exports.paginateListDocuments = paginateListDocuments;
exports.paginateListGroups = paginateListGroups;
exports.paginateListIndices = paginateListIndices;
exports.paginateListMessages = paginateListMessages;
exports.paginateListPluginActions = paginateListPluginActions;
exports.paginateListPluginTypeActions = paginateListPluginTypeActions;
exports.paginateListPluginTypeMetadata = paginateListPluginTypeMetadata;
exports.paginateListPlugins = paginateListPlugins;
exports.paginateListRetrievers = paginateListRetrievers;
exports.paginateListSubscriptions = paginateListSubscriptions;
exports.paginateListWebExperiences = paginateListWebExperiences;
exports.paginateSearchRelevantContent = paginateSearchRelevantContent;
